<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started with BioYourOwnBowl • BioYourOwnBowl</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/journal/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with BioYourOwnBowl">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioYourOwnBowl</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioYourOwnBowl-cnmf.html">Consensus Non-negative Matrix Factorization (cNMF) Results Exploration</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-main.html">Getting Started with BioYourOwnBowl</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-vis.html">Customized Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started with BioYourOwnBowl</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/blob/HEAD/vignettes/BioYourOwnBowl-main.Rmd"><code>vignettes/BioYourOwnBowl-main.Rmd</code></a></small>
      <div class="hidden name"><code>BioYourOwnBowl-main.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><strong>BioYourOwnBowl</strong> provides a comprehensive pipeline for
single-cell RNA-seq analysis using Seurat. This vignette demonstrates
the main workflow using example data from article: <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8376199/" class="external-link uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC8376199/</a>. To
maximize the efficiency, this object is down-sampled and has metadata
included.</p>
<p>The results from this notebook can be found in <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/tree/main/vignettes/main_workflow" class="uri">https://github.com/june-zhang-bioinfo/BioYourOwnBowl/tree/main/vignettes/main_workflow</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"june-zhang-bioinfo/BioYourOwnBowl"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl">BioYourOwnBowl</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; 'SeuratObject' was built under R 4.4.0 but the current version is</span></span>
<span><span class="co">#&gt; 4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI</span></span>
<span><span class="co">#&gt; for R may have changed</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'SeuratObject'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/circlize" class="external-link">circlize</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; circlize version 0.4.16</span></span>
<span><span class="co">#&gt; CRAN page: https://cran.r-project.org/package=circlize</span></span>
<span><span class="co">#&gt; Github page: https://github.com/jokergoo/circlize</span></span>
<span><span class="co">#&gt; Documentation: https://jokergoo.github.io/circlize_book/book/</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span><span class="co">#&gt; Gu, Z. circlize implements and enhances circular visualization</span></span>
<span><span class="co">#&gt;   in R. Bioinformatics 2014.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; This message can be suppressed by:</span></span>
<span><span class="co">#&gt;   suppressPackageStartupMessages(library(circlize))</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jolars/eulerr" class="external-link">eulerr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">treemap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grDevices</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">utils</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glue.tidyverse.org/" class="external-link">glue</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jeroen.r-universe.dev/jsonlite" class="external-link">jsonlite</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Matrix'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:tidyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand, pack, unpack</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/immunogenomics/harmony" class="external-link">harmony</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: Rcpp</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; ComplexHeatmap version 2.25.2</span></span>
<span><span class="co">#&gt; Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span><span class="co">#&gt; Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span><span class="co">#&gt; Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite either one:</span></span>
<span><span class="co">#&gt; - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span><span class="co">#&gt; - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span><span class="co">#&gt;     genomic data. Bioinformatics 2016.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The new InteractiveComplexHeatmap package can directly export static </span></span>
<span><span class="co">#&gt; complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; This message can be suppressed by:</span></span>
<span><span class="co">#&gt;   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'pheatmap'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ComplexHeatmap':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     pheatmap</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://anndata.dynverse.org" class="external-link">anndata</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'anndata'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:SeuratObject':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Layers</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://research.fredhutch.org/setty/en.html" class="external-link">convert2anndata</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/samuel-marsh/scCustomize" class="external-link">scCustomize</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; scCustomize v3.2.2</span></span>
<span><span class="co">#&gt; If you find the scCustomize useful please cite.</span></span>
<span><span class="co">#&gt; See 'samuel-marsh.github.io/scCustomize/articles/FAQ.html' for citation info.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rlang.r-lib.org" class="external-link">rlang</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'rlang'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:jsonlite':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     flatten, unbox</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/HenrikBengtsson/matrixStats" class="external-link">matrixStats</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'matrixStats'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     count</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">sessioninfo</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vubiostat/r-yaml/" class="external-link">yaml</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'yaml' was built under R version 4.4.3</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-example-data">Prepare Example Data<a class="anchor" aria-label="anchor" href="#prepare-example-data"></a>
</h2>
<p>We’ll start with loading example dataset.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_dir_base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"BioYourOwnBowl"</span><span class="op">)</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"GSE178756_downsampled.rds"</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">data_path</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 18798 features across 10000 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (18798 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  1 layer present: counts</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-color-palette">Define Color Palette<a class="anchor" aria-label="anchor" href="#define-color-palette"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"#e6194B"</span>, </span>
<span>  <span class="st">"#86D3B2"</span>, </span>
<span>  <span class="st">"#5E8DBA"</span>, </span>
<span>  <span class="st">"#F7A131"</span>, </span>
<span>  <span class="st">"#911eb4"</span>, </span>
<span>  <span class="st">"#46f0f0"</span>, </span>
<span>  <span class="st">"#fabebe"</span>, </span>
<span>  <span class="st">"#bcf60c"</span>, </span>
<span>  <span class="st">"#B88C6D"</span>,</span>
<span>  <span class="st">"#4d88e8"</span>, </span>
<span>  <span class="st">"#FFFB47"</span>, </span>
<span>  <span class="st">"#A1A74A"</span>,</span>
<span>  <span class="st">"#7BB1A4"</span>, </span>
<span>  <span class="st">"#b15928"</span>,</span>
<span>  <span class="st">"#C493D4"</span>, </span>
<span>  <span class="st">"#f58231"</span>,</span>
<span>  <span class="st">"#0072B2"</span>,</span>
<span>  <span class="st">"#D55E00"</span>, </span>
<span>  <span class="st">"#009E73"</span>,</span>
<span>  <span class="st">"#CC79A7"</span>,</span>
<span>  <span class="st">"#F0E442"</span> </span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-selected-features">Define Selected Features<a class="anchor" aria-label="anchor" href="#define-selected-features"></a>
</h2>
<p>You can provide genes of interest. These genes will be used for
Highly variable genes information (HVG_info.csv) and main-variance plot
(so you can see if they are considered as variable features) and the dot
plot for your clustering results.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"CD8A"</span>, <span class="st">"CD8B"</span>,</span>
<span>  <span class="st">"CD3D"</span>, <span class="st">"CD3E"</span>, <span class="st">"CD3G"</span>,</span>
<span>  <span class="st">"GZMB"</span>, <span class="st">"GZMA"</span>, <span class="st">"GZMH"</span>,</span>
<span>  <span class="st">"PRF1"</span>, <span class="st">"NKG7"</span>, <span class="st">"GNLY"</span>,</span>
<span>  <span class="st">"IL2RA"</span>, <span class="st">"TNFRSF9"</span>, <span class="st">"CD69"</span>,</span>
<span>  <span class="st">"MKI67"</span>, <span class="st">"TOP2A"</span>,</span>
<span>  <span class="st">"PDCD1"</span>, <span class="st">"LAG3"</span>, <span class="st">"TIGIT"</span>,</span>
<span>  <span class="st">"CTLA4"</span>, <span class="st">"HAVCR2"</span>,</span>
<span>  <span class="st">"CCR7"</span>, <span class="st">"SELL"</span>, <span class="st">"TCF7"</span>,</span>
<span>  <span class="st">"LEF1"</span>, <span class="st">"IL7R"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-pipeline">Run Pipeline<a class="anchor" aria-label="anchor" href="#run-pipeline"></a>
</h2>
<p>The <code><a href="../reference/optimize_single_cell.html">optimize_single_cell()</a></code> function performs
comprehensive single-cell analysis:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/optimize_single_cell.html">optimize_single_cell</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span></span>
<span>  <span class="co"># Normalization parameters</span></span>
<span>  normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>  vf.method <span class="op">=</span> <span class="st">"vst"</span>,</span>
<span>  model.use <span class="op">=</span> <span class="st">"linear"</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">2000</span>,</span>
<span></span>
<span>  <span class="co"># Quality control</span></span>
<span>  mt.cutoff <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  gene_filter <span class="op">=</span> <span class="fl">100</span>,</span>
<span></span>
<span>  <span class="co"># Clustering parameters</span></span>
<span>  k_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span><span class="op">)</span>,</span>
<span>  resolution_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  clusters_min <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  clusters_max <span class="op">=</span> <span class="fl">9</span>,</span>
<span></span>
<span>  <span class="co"># UMAP parameters</span></span>
<span>  min.dist <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">17</span>,</span>
<span></span>
<span>  <span class="co"># Regression (optional)</span></span>
<span>  regress.out <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span></span>
<span>  <span class="co"># Visualization</span></span>
<span>  colors <span class="op">=</span> <span class="va">colors</span>,</span>
<span>  cluster_distribution_layers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Time_point"</span>, <span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>  cluster_distribution_layer_orders <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time_point <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Post_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span></span>
<span>  <span class="co"># Selected features for visualization</span></span>
<span>  features <span class="op">=</span> <span class="va">selected_features</span>,</span>
<span></span>
<span>  <span class="co"># Output directory</span></span>
<span>  out_dir <span class="op">=</span> <span class="st">"/Users/zz005/Documents/github/R/vignettes_results"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in dir.create(out_dir):</span></span>
<span><span class="co">#&gt; '/Users/zz005/Documents/github/R/vignettes_results' already exists</span></span>
<span><span class="co">#&gt; parameters saved to logs/params_20251216_162836.yaml</span></span>
<span><span class="co">#&gt; Session info saved to logs/session_info_20251216_162836.txt</span></span>
<span><span class="co">#&gt; Gene filtering done.</span></span>
<span><span class="co">#&gt; Normalizing layer: counts</span></span>
<span><span class="co">#&gt; Finding variable features for layer counts</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Preprocess object done.</span></span>
<span><span class="co">#&gt; Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span></span>
<span><span class="co">#&gt; To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span></span>
<span><span class="co">#&gt; This message will be shown once per session</span></span>
<span><span class="co">#&gt; 16:28:46 UMAP embedding parameters a = 0.9922 b = 1.112</span></span>
<span><span class="co">#&gt; 16:28:46 Read 10000 rows and found 30 numeric columns</span></span>
<span><span class="co">#&gt; 16:28:46 Using Annoy for neighbor search, n_neighbors = 30</span></span>
<span><span class="co">#&gt; 16:28:46 Building Annoy index with metric = cosine, n_trees = 50</span></span>
<span><span class="co">#&gt; 0%   10   20   30   40   50   60   70   80   90   100%</span></span>
<span><span class="co">#&gt; [----|----|----|----|----|----|----|----|----|----|</span></span>
<span><span class="co">#&gt; **************************************************|</span></span>
<span><span class="co">#&gt; 16:28:46 Writing NN index file to temp file /var/folders/tw/783hgvc55jv1b7mqflwpjc7h0000gp/T//RtmpSm6IMd/file1246e65789bed</span></span>
<span><span class="co">#&gt; 16:28:46 Searching Annoy index using 1 thread, search_k = 3000</span></span>
<span><span class="co">#&gt; 16:28:48 Annoy recall = 100%</span></span>
<span><span class="co">#&gt; 16:28:48 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30</span></span>
<span><span class="co">#&gt; 16:28:48 Initializing from normalized Laplacian + noise (using RSpectra)</span></span>
<span><span class="co">#&gt; 16:28:49 Commencing optimization for 500 epochs, with 470206 positive edges</span></span>
<span><span class="co">#&gt; 16:28:49 Using rng type: pcg</span></span>
<span><span class="co">#&gt; 16:28:58 Optimization finished</span></span>
<span><span class="co">#&gt; Computing nearest neighbor graph</span></span>
<span><span class="co">#&gt; Computing SNN</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 10000</span></span>
<span><span class="co">#&gt; Number of edges: 356401</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.7500</span></span>
<span><span class="co">#&gt; Number of communities: 25</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `FetchData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Removing clusters: 18</span></span>
<span><span class="co">#&gt; Warning: `PackageCheck()` was deprecated in SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `rlang::check_installed()` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Picking joint bandwidth of 99.9</span></span>
<span><span class="co">#&gt; Gene filtering done.</span></span>
<span><span class="co">#&gt; Normalizing layer: counts</span></span>
<span><span class="co">#&gt; Finding variable features for layer counts</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Preprocess object done.</span></span>
<span><span class="co">#&gt; Warning in plot_mean_variance(object = object, features = features): The</span></span>
<span><span class="co">#&gt; following selected features were not found in the object: TNFRSF9, MKI67, TOP2A</span></span>
<span><span class="co">#&gt; Plot Mean-Variance done.</span></span>
<span><span class="co">#&gt; PC_ 1 </span></span>
<span><span class="co">#&gt; Positive:  TRAC, CCL5, TRBC2, NKG7, GZMA, CTSW, LTB, IL7R, CST7, GZMH </span></span>
<span><span class="co">#&gt;     CD8B, KLRK1, KLRD1, CMC1, C12orf75, HOPX, CD7, KLRG1, TRGC2, GZMK </span></span>
<span><span class="co">#&gt;     PRF1, GNLY, CD8A, CCL4, CD27, FGFBP2, LDHB, CD160, SYNE2, RHOH </span></span>
<span><span class="co">#&gt; Negative:  FCN1, CYBB, SERPINA1, MNDA, IFI30, MPEG1, CST3, CPVL, CLEC7A, CD14 </span></span>
<span><span class="co">#&gt;     LYZ, VCAN, SPI1, CD68, NCF2, MS4A6A, CSTA, FCER1G, PLBD1, S100A9 </span></span>
<span><span class="co">#&gt;     KLF4, SLC7A7, LILRB2, HCK, CD36, FCGR2A, MARCKS, S100A8, S100A12, FGL2 </span></span>
<span><span class="co">#&gt; PC_ 2 </span></span>
<span><span class="co">#&gt; Positive:  JUND, JUN, KLF2, ZFP36L2, CD69, DNAJB1, IL7R, H1FX, TSC22D3, CXCR4 </span></span>
<span><span class="co">#&gt;     STK17B, BTG1, TPT1, JUNB, FOS, DUSP1, TCF7, ARL4C, BTG2, TNFAIP3 </span></span>
<span><span class="co">#&gt;     LEPROTL1, FOXP1, MTRNR2L12, TMEM123, DUSP2, RCAN3, PPP1R15A, TOB1, LDHB, PABPC1 </span></span>
<span><span class="co">#&gt; Negative:  GZMH, NKG7, CCL5, FGFBP2, HLA.DRB1, FCGR3A, KLRD1, TYROBP, CD160, CMC1 </span></span>
<span><span class="co">#&gt;     FGR, GZMA, HLA.DPB1, SERPINA1, LYZ, GZMB, CSTA, GNLY, FCN1, HLA.DPA1 </span></span>
<span><span class="co">#&gt;     PRF1, CST3, S100A9, CST7, MNDA, IFI30, S100A8, HOPX, CD68, CLEC7A </span></span>
<span><span class="co">#&gt; PC_ 3 </span></span>
<span><span class="co">#&gt; Positive:  NKG7, CST7, CCL5, GZMA, GZMH, CTSW, KLRD1, PRF1, GZMB, GNLY </span></span>
<span><span class="co">#&gt;     FGFBP2, CCL4, TRGC2, PLEK, KLRK1, EFHD2, HOPX, PFN1, C12orf75, CMC1 </span></span>
<span><span class="co">#&gt;     CD8A, KLRG1, ZEB2, HLA.DPB1, LYAR, SRGN, CX3CR1, S100A4, MATK, ADGRG1 </span></span>
<span><span class="co">#&gt; Negative:  LTB, SERINC5, RCAN3, CCR7, LEF1, TSHZ2, MAL, MAML2, IL6ST, IL7R </span></span>
<span><span class="co">#&gt;     PRKCA, FHIT, TRABD2A, ACTN1, TXK, FOXP1, BEX3, BCL2, AIF1, SELL </span></span>
<span><span class="co">#&gt;     NFKBIZ, IL6R, PIK3IP1, TNFRSF25, TIAM1, ADTRP, TCF7, MAP3K1, LRRN3, MTRNR2L12 </span></span>
<span><span class="co">#&gt; PC_ 4 </span></span>
<span><span class="co">#&gt; Positive:  SYNE2, AHNAK, UTRN, HIST1H1E, VPS13C, SYNE1, ATF7IP, HIST1H1D, GBP5, KIAA1109 </span></span>
<span><span class="co">#&gt;     NIN, SLFN5, TRIP12, PIK3R1, MYBL1, GOLGB1, SETX, CEP350, SETD2, ITGA4 </span></span>
<span><span class="co">#&gt;     GOLGA4, CD84, ANKRD36C, DOCK10, GPRIN3, PYHIN1, BOD1L1, ZEB2, PCNX1, IQGAP1 </span></span>
<span><span class="co">#&gt; Negative:  ACTB, EEF1A1, PFN1, FXYD5, LDHB, FTH1, IFITM1, FTL, GAPDH, ACTG1 </span></span>
<span><span class="co">#&gt;     TRAC, EIF3E, EIF3L, GABARAP, NOSIP, CD7, HSPB1, GYPC, TRBC2, S100A6 </span></span>
<span><span class="co">#&gt;     CD27, FLT3LG, RGS10, PPA1, UCP2, ATP6V0C, FKBP1A, SLC25A5, PKM, PLP2 </span></span>
<span><span class="co">#&gt; PC_ 5 </span></span>
<span><span class="co">#&gt; Positive:  S100A11, LTB, CRIP1, PLP2, LDHB, TRAC, S100A10, S100A4, S100A6, VIM </span></span>
<span><span class="co">#&gt;     TNFSF10, NAP1L1, CAPG, ITGB1, SIT1, TRAT1, EEF1A1, RNASET2, CISH, TNFRSF4 </span></span>
<span><span class="co">#&gt;     CD82, CXCR3, STAT1, MAL, LGALS1, PKM, IL7R, FXYD5, TSPO, SUSD3 </span></span>
<span><span class="co">#&gt; Negative:  HSPA1B, HSPA1A, NR4A1, EGR1, RHOB, DUSP2, DUSP1, DNAJB1, ZFP36, HSPA6 </span></span>
<span><span class="co">#&gt;     MTRNR2L12, JUN, DNAJB4, PMAIP1, CD69, FOS, CSRNP1, NR4A2, RGS2, IER3 </span></span>
<span><span class="co">#&gt;     BTG2, TNF, NFKBIA, SERTAD1, DDIT3, JUND, NFKBIZ, ATF3, IER5, PPP1R15A </span></span>
<span><span class="co">#&gt; Top genes per PC exported to PCs_top-genes.csv</span></span>
<span><span class="co">#&gt; Elbow plot saved to /Users/zz005/Documents/github/R/vignettes_results/elbow.png</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-main_files/figure-html/unnamed-chunk-6-1.png" class="r-plt" width="500"></p>
<pre><code><span><span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14</span></span>
<span><span class="co">#&gt; 16:29:08 UMAP embedding parameters a = 0.9922 b = 1.112</span></span>
<span><span class="co">#&gt; 16:29:08 Read 5893 rows and found 14 numeric columns</span></span>
<span><span class="co">#&gt; 16:29:08 Using Annoy for neighbor search, n_neighbors = 30</span></span>
<span><span class="co">#&gt; 16:29:08 Building Annoy index with metric = cosine, n_trees = 50</span></span>
<span><span class="co">#&gt; 0%   10   20   30   40   50   60   70   80   90   100%</span></span>
<span><span class="co">#&gt; [----|----|----|----|----|----|----|----|----|----|</span></span>
<span><span class="co">#&gt; **************************************************|</span></span>
<span><span class="co">#&gt; 16:29:08 Writing NN index file to temp file /var/folders/tw/783hgvc55jv1b7mqflwpjc7h0000gp/T//RtmpSm6IMd/file1246e24c7745b</span></span>
<span><span class="co">#&gt; 16:29:08 Searching Annoy index using 1 thread, search_k = 3000</span></span>
<span><span class="co">#&gt; 16:29:09 Annoy recall = 100%</span></span>
<span><span class="co">#&gt; 16:29:09 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30</span></span>
<span><span class="co">#&gt; 16:29:10 Initializing from normalized Laplacian + noise (using RSpectra)</span></span>
<span><span class="co">#&gt; 16:29:10 Commencing optimization for 500 epochs, with 245502 positive edges</span></span>
<span><span class="co">#&gt; 16:29:10 Using rng type: pcg</span></span>
<span><span class="co">#&gt; 16:29:15 Optimization finished</span></span>
<span><span class="co">#&gt; ---- Testing k.param = 10 ----</span></span>
<span><span class="co">#&gt; Computing nearest neighbor graph</span></span>
<span><span class="co">#&gt; Computing SNN</span></span>
<span><span class="co">#&gt; Running: k.param = 10 | resolution = 0.1</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 5893</span></span>
<span><span class="co">#&gt; Number of edges: 86275</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9618</span></span>
<span><span class="co">#&gt; Number of communities: 6</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; ✅ Done: k.param = 10 | resolution = 0.1</span></span>
<span><span class="co">#&gt; Running: k.param = 10 | resolution = 0.15</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 5893</span></span>
<span><span class="co">#&gt; Number of edges: 86275</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9522</span></span>
<span><span class="co">#&gt; Number of communities: 8</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; ✅ Done: k.param = 10 | resolution = 0.15</span></span>
<span><span class="co">#&gt; Running: k.param = 10 | resolution = 0.2</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 5893</span></span>
<span><span class="co">#&gt; Number of edges: 86275</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9444</span></span>
<span><span class="co">#&gt; Number of communities: 10</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; Too many clusters (10); skipping higher resolutions.</span></span>
<span><span class="co">#&gt; ---- Testing k.param = 20 ----</span></span>
<span><span class="co">#&gt; Computing nearest neighbor graph</span></span>
<span><span class="co">#&gt; Computing SNN</span></span>
<span><span class="co">#&gt; Running: k.param = 20 | resolution = 0.1</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 5893</span></span>
<span><span class="co">#&gt; Number of edges: 213559</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9594</span></span>
<span><span class="co">#&gt; Number of communities: 7</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; ✅ Done: k.param = 20 | resolution = 0.1</span></span>
<span><span class="co">#&gt; Running: k.param = 20 | resolution = 0.15</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 5893</span></span>
<span><span class="co">#&gt; Number of edges: 213559</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9474</span></span>
<span><span class="co">#&gt; Number of communities: 7</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; ✅ Done: k.param = 20 | resolution = 0.15</span></span>
<span><span class="co">#&gt; Running: k.param = 20 | resolution = 0.2</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 5893</span></span>
<span><span class="co">#&gt; Number of edges: 213559</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9380</span></span>
<span><span class="co">#&gt; Number of communities: 9</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; ✅ Done: k.param = 20 | resolution = 0.2</span></span>
<span><span class="co">#&gt; All clustering tests completed. Results saved to dimplot.pdf</span></span>
<span><span class="co">#&gt; [1] "k10_r0.1"  "k10_r0.15" "k20_r0.1"  "k20_r0.15" "k20_r0.2" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    0    1    2    3    4    5 </span></span>
<span><span class="co">#&gt; 2538 1306 1174  390  284  201</span></span>
<span><span class="co">#&gt; Calculating cluster 0</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `GetAssayData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; For a (much!) faster implementation of the Wilcoxon Rank Sum Test,</span></span>
<span><span class="co">#&gt; (default method for FindMarkers) please install the presto package</span></span>
<span><span class="co">#&gt; --------------------------------------------</span></span>
<span><span class="co">#&gt; install.packages('devtools')</span></span>
<span><span class="co">#&gt; devtools::install_github('immunogenomics/presto')</span></span>
<span><span class="co">#&gt; --------------------------------------------</span></span>
<span><span class="co">#&gt; After installation of presto, Seurat will automatically use the more </span></span>
<span><span class="co">#&gt; efficient implementation (no further action necessary).</span></span>
<span><span class="co">#&gt; This message will be shown once per session</span></span>
<span><span class="co">#&gt; Calculating cluster 1</span></span>
<span><span class="co">#&gt; Calculating cluster 2</span></span>
<span><span class="co">#&gt; Calculating cluster 3</span></span>
<span><span class="co">#&gt; Calculating cluster 4</span></span>
<span><span class="co">#&gt; Calculating cluster 5</span></span>
<span><span class="co">#&gt; Warning: The following requested variables were not found: TNFRSF9, MKI67,</span></span>
<span><span class="co">#&gt; TOP2A</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Automatically calculated height: 9.2 inches for 150 genes</span></span>
<span><span class="co">#&gt; `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">#&gt; 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">#&gt; setting TRUE/FALSE to it.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; Heatmap saved to k10_r0.1_heatmap.png</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    0    1    2    3    4    5    6    7 </span></span>
<span><span class="co">#&gt; 1447 1285 1213  785  424  284  254  201</span></span>
<span><span class="co">#&gt; Calculating cluster 0</span></span>
<span><span class="co">#&gt; Calculating cluster 1</span></span>
<span><span class="co">#&gt; Calculating cluster 2</span></span>
<span><span class="co">#&gt; Calculating cluster 3</span></span>
<span><span class="co">#&gt; Calculating cluster 4</span></span>
<span><span class="co">#&gt; Calculating cluster 5</span></span>
<span><span class="co">#&gt; Calculating cluster 6</span></span>
<span><span class="co">#&gt; Calculating cluster 7</span></span>
<span><span class="co">#&gt; Warning: The following requested variables were not found: TNFRSF9, MKI67,</span></span>
<span><span class="co">#&gt; TOP2A</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Automatically calculated height: 10.84 inches for 191 genes</span></span>
<span><span class="co">#&gt; `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">#&gt; 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">#&gt; setting TRUE/FALSE to it.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; Heatmap saved to k10_r0.15_heatmap.png</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    0    1    2    3    4    5    6 </span></span>
<span><span class="co">#&gt; 2558 1292 1172  396  218  199   58</span></span>
<span><span class="co">#&gt; Calculating cluster 0</span></span>
<span><span class="co">#&gt; Calculating cluster 1</span></span>
<span><span class="co">#&gt; Calculating cluster 2</span></span>
<span><span class="co">#&gt; Calculating cluster 3</span></span>
<span><span class="co">#&gt; Calculating cluster 4</span></span>
<span><span class="co">#&gt; Calculating cluster 5</span></span>
<span><span class="co">#&gt; Calculating cluster 6</span></span>
<span><span class="co">#&gt; Warning: The following requested variables were not found: TNFRSF9, MKI67,</span></span>
<span><span class="co">#&gt; TOP2A</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Automatically calculated height: 10.2 inches for 175 genes</span></span>
<span><span class="co">#&gt; `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">#&gt; 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">#&gt; setting TRUE/FALSE to it.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; Heatmap saved to k20_r0.1_heatmap.png</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    0    1    2    3    4    5    6 </span></span>
<span><span class="co">#&gt; 1645 1302 1299 1172  218  199   58</span></span>
<span><span class="co">#&gt; Calculating cluster 0</span></span>
<span><span class="co">#&gt; Calculating cluster 1</span></span>
<span><span class="co">#&gt; Calculating cluster 2</span></span>
<span><span class="co">#&gt; Calculating cluster 3</span></span>
<span><span class="co">#&gt; Calculating cluster 4</span></span>
<span><span class="co">#&gt; Calculating cluster 5</span></span>
<span><span class="co">#&gt; Calculating cluster 6</span></span>
<span><span class="co">#&gt; Warning: The following requested variables were not found: TNFRSF9, MKI67,</span></span>
<span><span class="co">#&gt; TOP2A</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Automatically calculated height: 10.08 inches for 172 genes</span></span>
<span><span class="co">#&gt; `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">#&gt; 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">#&gt; setting TRUE/FALSE to it.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; Heatmap saved to k20_r0.15_heatmap.png</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    0    1    2    3    4    5    6    7    8 </span></span>
<span><span class="co">#&gt; 1389 1300 1172  882  426  249  218  199   58</span></span>
<span><span class="co">#&gt; Calculating cluster 0</span></span>
<span><span class="co">#&gt; Calculating cluster 1</span></span>
<span><span class="co">#&gt; Calculating cluster 2</span></span>
<span><span class="co">#&gt; Calculating cluster 3</span></span>
<span><span class="co">#&gt; Calculating cluster 4</span></span>
<span><span class="co">#&gt; Calculating cluster 5</span></span>
<span><span class="co">#&gt; Calculating cluster 6</span></span>
<span><span class="co">#&gt; Calculating cluster 7</span></span>
<span><span class="co">#&gt; Calculating cluster 8</span></span>
<span><span class="co">#&gt; Warning: The following requested variables were not found: TNFRSF9, MKI67,</span></span>
<span><span class="co">#&gt; TOP2A</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Automatically calculated height: 11.84 inches for 216 genes</span></span>
<span><span class="co">#&gt; `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">#&gt; 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">#&gt; setting TRUE/FALSE to it.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; Heatmap saved to k20_r0.2_heatmap.png</span></span>
<span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 9029 features across 5893 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (9029 features, 2000 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  2 dimensional reductions calculated: pca, umap</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="understanding-the-parameters">Understanding the Parameters<a class="anchor" aria-label="anchor" href="#understanding-the-parameters"></a>
</h2>
<div class="section level3">
<h3 id="normalization-feature-selection">Normalization &amp; Feature Selection<a class="anchor" aria-label="anchor" href="#normalization-feature-selection"></a>
</h3>
<ul>
<li>
<strong>vf.method</strong>: Method for variable feature selection
<ul>
<li>
<code>"vst"</code>: Variance stabilizing transformation
(default)</li>
<li>
<code>"dispersion"</code>: Dispersion-based selection</li>
<li>
<code>"pseudobulk"</code>: Pseudobulk aggregation</li>
<li>
<code>"sct"</code>: SCTransform</li>
<li>
<code>"anchor"</code>: Integration anchors</li>
</ul>
</li>
<li>
<strong>nfeatures</strong>: Number of variable features to select
(default: 2000)</li>
</ul>
</div>
<div class="section level3">
<h3 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h3>
<ul>
<li>
<strong>mt.cutoff</strong>: Maximum mitochondrial content percentage
(default: 25)</li>
<li>
<strong>gene_filter</strong>: Minimum cells a gene must be expressed
in (default: 50)</li>
</ul>
</div>
<div class="section level3">
<h3 id="clustering-optimization">Clustering Optimization<a class="anchor" aria-label="anchor" href="#clustering-optimization"></a>
</h3>
<p>The function tests multiple clustering parameters:</p>
<ul>
<li>
<strong>k_range</strong>: Neighbor counts for graph
construction</li>
<li>
<strong>resolution_range</strong>: Resolution values for Louvain
clustering</li>
<li>
<strong>clusters_min/max</strong>: Expected number of clusters</li>
</ul>
</div>
<div class="section level3">
<h3 id="output-files">Output Files<a class="anchor" aria-label="anchor" href="#output-files"></a>
</h3>
<p>The pipeline generates:</p>
<ul>
<li>
<code>logs/</code> - Parameters and session info</li>
<li>
<code>qc.pdf</code> - Quality control plots</li>
<li>
<code>HVG_info.csv</code> - The ranking of variable features and the
number of cells express them</li>
<li>
<code>elbow.png</code> - PC selection plot</li>
<li>
<code>Mean-variance_*.png</code> - Variable feature plots</li>
<li>
<code>PCs_top-genes.csv</code> - Top genes per principal
component</li>
<li>
<code>dimplot.pdf</code> - Dimplots showing clustering results</li>
<li>
<code>k*_r*_barplot.png</code> - Cluster distribution plots</li>
<li>
<code>k*_r*_dotplot.pdf</code> - Dot plots for each clustering,
including selected features and DGE</li>
<li>
<code>k*_r*_heatmap.png</code> - Expression heatmaps for DGE</li>
<li>
<code>object.rds</code> - Final processed Seurat object</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="customize-your-own-pipeline">Customize your own pipeline<a class="anchor" aria-label="anchor" href="#customize-your-own-pipeline"></a>
</h2>
<div class="section level3">
<h3 id="quality-control-1">Quality control<a class="anchor" aria-label="anchor" href="#quality-control-1"></a>
</h3>
<p>n* are used to trace the dimension changes over the steps.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define your variables</span></span>
<span><span class="va">output_dir</span> <span class="op">&lt;-</span> <span class="st">"your/path"</span></span>
<span><span class="va">gene_filter</span> <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="va">vf.method</span> <span class="op">=</span> <span class="st">"dispersion"</span></span>
<span><span class="va">regress.out</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MKI67"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"qc.pdf"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">n1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_genes_by_expression.html">filter_genes_by_expression</a></span><span class="op">(</span><span class="va">object</span>, gene_filter <span class="op">=</span> <span class="va">gene_filter</span>, assay <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span><span class="va">n2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># cluster-based cell filtering</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess_obj.html">preprocess_obj</a></span><span class="op">(</span><span class="va">object</span>,</span>
<span>                         vf.method <span class="op">=</span> <span class="va">vf.method</span>,</span>
<span>                         nfeatures <span class="op">=</span> <span class="va">nfeatures</span>,</span>
<span>                         regress.out <span class="op">=</span> <span class="va">regress.out</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">object</span>, npcs <span class="op">=</span> <span class="fl">30</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">object</span>, reduction <span class="op">=</span> <span class="va">reduction</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, seed.use <span class="op">=</span> <span class="va">seed</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">object</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">object</span>, resolution <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">object</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, label.box <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">object</span>, features <span class="op">=</span> <span class="st">"percent.mt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">object</span>, features <span class="op">=</span> <span class="st">"nCount_RNA"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">n4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_low_quality_clusters.html">remove_low_quality_clusters</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">object</span>, <span class="va">percent.mt</span> <span class="op">&lt;</span> <span class="fl">25</span><span class="op">)</span></span>
<span></span>
<span><span class="va">n5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.newpage.html" class="external-link">grid.newpage</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html" class="external-link">grid.draw</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Cells before QC:"</span>, <span class="va">n4</span>, <span class="st">"\n"</span>,</span>
<span>         <span class="st">"Cells after QC:"</span>, <span class="va">n5</span><span class="op">)</span>,</span>
<span>  x <span class="op">=</span> <span class="fl">0.5</span>, y <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_genes_by_expression.html">filter_genes_by_expression</a></span><span class="op">(</span><span class="va">object</span>, gene_filter <span class="op">=</span> <span class="va">gene_filter</span><span class="op">)</span></span>
<span><span class="va">n3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html" class="external-link">grid.draw</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Genes before filtering:"</span>, <span class="va">n1</span>, <span class="st">"\n"</span>,</span>
<span>         <span class="st">"Genes after 1st filtering:"</span>, <span class="va">n2</span>, <span class="st">"\n"</span>,</span>
<span>         <span class="st">"Genes after 2nd filtering"</span>, <span class="va">n3</span><span class="op">)</span>,</span>
<span>  x <span class="op">=</span> <span class="fl">0.5</span>, y <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="if-you-prefer-hard-cutoff-you-can-skip-the-last-chunk-and-do-this">If you prefer hard cutoff, you can skip the last chunk and do
this:<a class="anchor" aria-label="anchor" href="#if-you-prefer-hard-cutoff-you-can-skip-the-last-chunk-and-do-this"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"qc.pdf"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># gene filtering</span></span>
<span><span class="va">n1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_genes_by_expression.html">filter_genes_by_expression</a></span><span class="op">(</span><span class="va">object</span>, gene_filter <span class="op">=</span> <span class="va">gene_filter</span>, assay <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span><span class="va">n2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># choose cutoff based on violin plot</span></span>
<span><span class="fu">Vlnplot</span><span class="op">(</span><span class="va">object</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"nFeature_RNA"</span>, <span class="st">"nCount_RNA"</span>, <span class="st">"percent.mt"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">n3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">object</span>, <span class="va">nFeature_RNA</span> <span class="op">&gt;</span> <span class="fl">100</span> <span class="op">&amp;</span> <span class="va">nCount_RNA</span> <span class="op">&lt;</span> <span class="fl">2000</span> <span class="op">&amp;</span> <span class="va">percent.mt</span> <span class="op">&lt;</span> <span class="fl">25</span><span class="op">)</span></span>
<span><span class="va">n4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html" class="external-link">grid.draw</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Cells before QC:"</span>, <span class="va">n3</span>, <span class="st">"\n"</span>,</span>
<span>         <span class="st">"Cells after QC:"</span>, <span class="va">n4</span><span class="op">)</span>,</span>
<span>  x <span class="op">=</span> <span class="fl">0.5</span>, y <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_genes_by_expression.html">filter_genes_by_expression</a></span><span class="op">(</span><span class="va">object</span>, gene_filter <span class="op">=</span> <span class="va">gene_filter</span>, assay <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span><span class="va">n5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.newpage.html" class="external-link">grid.newpage</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html" class="external-link">grid.draw</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Genes before filtering:"</span>, <span class="va">n1</span>, <span class="st">"\n"</span>,</span>
<span>         <span class="st">"Genes after 1st filtering"</span>, <span class="va">n2</span>, <span class="st">"\n"</span>,</span>
<span>         <span class="st">"Genes after 2nd filtering"</span>, <span class="va">n5</span><span class="op">)</span>,</span>
<span>  x <span class="op">=</span> <span class="fl">0.5</span>, y <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="preprocess-and-dimensional-reduction">Preprocess and dimensional reduction<a class="anchor" aria-label="anchor" href="#preprocess-and-dimensional-reduction"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess_obj.html">preprocess_obj</a></span><span class="op">(</span><span class="va">object</span>,</span>
<span>                         vf.method <span class="op">=</span> <span class="va">vf.method</span>,</span>
<span>                         nfeatures <span class="op">=</span> <span class="va">nfeatures</span>,</span>
<span>                         regress.out <span class="op">=</span> <span class="va">regress.out</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># export gene rank - conditionally applicable</span></span>
<span><span class="va">hvf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">HVFInfo</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span>
<span><span class="va">hvf</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hvf</span><span class="op">)</span></span>
<span><span class="va">hvf</span> <span class="op">&lt;-</span> <span class="va">hvf</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">hvf</span><span class="op">$</span><span class="va">mvp.dispersion</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">hvf</span><span class="op">$</span><span class="va">rank</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">hvf</span><span class="op">)</span></span>
<span><span class="va">hvf</span><span class="op">$</span><span class="va">features</span> <span class="op">&lt;-</span> <span class="st">"No"</span></span>
<span><span class="va">hvf</span><span class="op">$</span><span class="va">features</span><span class="op">[</span><span class="va">hvf</span><span class="op">$</span><span class="va">gene</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">features</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Yes"</span></span>
<span><span class="va">expr_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">object</span>, layer <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="va">gene_cell_counts</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">expr_matrix</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">hvf</span><span class="op">$</span><span class="va">num_cells_expressed</span> <span class="op">&lt;-</span> <span class="va">gene_cell_counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">hvf</span><span class="op">$</span><span class="va">gene</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene_cell_counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">hvf</span>, <span class="st">"HVG_info.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># generate mean-variance plot</span></span>
<span><span class="fu"><a href="../reference/plot_mean_variance.html">plot_mean_variance</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">object</span>,</span>
<span>  features <span class="op">=</span> <span class="va">selected_features</span></span>
<span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">object</span>, npcs <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># export top genes per PC</span></span>
<span><span class="fu"><a href="../reference/export_top_pc_genes.html">export_top_pc_genes</a></span><span class="op">(</span><span class="va">object</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, top_n <span class="op">=</span> <span class="fl">20</span>, n_pcs <span class="op">=</span> <span class="fl">30</span>, file_name <span class="op">=</span> <span class="st">"PCs_top-genes.csv"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">dims</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_PCs.html">select_PCs</a></span><span class="op">(</span><span class="va">object</span>,</span>
<span>                   improved_diff_quantile <span class="op">=</span> <span class="va">improved_diff_quantile</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">dims</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># You can also choose PCs based on what you prefer, according to top genes per PC.</span></span>
<span><span class="co"># e.g. you think PC3, PC4 and PC5 represent unwanted noise.</span></span>
<span><span class="va">dims</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">5</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="umap-clustering-and-dge">UMAP, clustering and DGE<a class="anchor" aria-label="anchor" href="#umap-clustering-and-dge"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">object</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="va">dims</span>, seed.use <span class="op">=</span> <span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clustering.html">clustering</a></span><span class="op">(</span><span class="va">object</span>, </span>
<span>                     colors <span class="op">=</span> <span class="va">colors</span>, </span>
<span>                     k_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">50</span>, <span class="fl">10</span><span class="op">)</span>, </span>
<span>                     resolution_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.4</span>, <span class="fl">0.05</span><span class="op">)</span>, </span>
<span>                     dims <span class="op">=</span> <span class="va">dims</span>, </span>
<span>                     clusters_min <span class="op">=</span> <span class="fl">6</span>, </span>
<span>                     clusters_max <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># see what k and r combinations meet your expectation</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^k\\d+_r\\d+\\.\\d+$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">object</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">temp_clusters</span> <span class="kw">in</span> <span class="va">clusters</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">temp_clusters</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stacked_bar_plots.html">stacked_bar_plots</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">object</span>,</span>
<span>    idents <span class="op">=</span> <span class="va">temp_clusters</span>,</span>
<span>    layers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Time_point"</span>, <span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>    layer_orders <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>      Time_point <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Post_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span></span>
<span>    <span class="op">)</span>,</span>
<span>    colors <span class="op">=</span> <span class="va">colors</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Cluster Distribution"</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">n_layers</span> <span class="op">&lt;-</span> <span class="va">object</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Time_point"</span>, <span class="st">"Cohort"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">as.character</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">width</span> <span class="op">&lt;-</span> <span class="fl">5</span> <span class="op">+</span> <span class="fl">0.3</span> <span class="op">*</span> <span class="va">n_layers</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">temp_clusters</span>, <span class="st">"_barplot.png"</span><span class="op">)</span>, plot <span class="op">=</span> <span class="va">plots</span>, width <span class="op">=</span> <span class="va">width</span>, height <span class="op">=</span> <span class="fl">5.5</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># DGE</span></span>
<span>  <span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">object</span>, min.pct <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span>  <span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_marker_genes_score.html">select_marker_genes_score</a></span><span class="op">(</span><span class="va">markers</span><span class="op">)</span></span>
<span></span>
<span>  <span class="fu"><a href="../reference/dotplots_pdf.html">dotplots_pdf</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">object</span>,</span>
<span>    features.1 <span class="op">=</span> <span class="va">features</span>,</span>
<span>    features.2 <span class="op">=</span> <span class="va">dge</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>    file_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">temp_clusters</span>, <span class="st">"_dotplot.pdf"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="co"># swap cell-level heatmap with pseodubulk-level</span></span>
<span>  <span class="fu"><a href="../reference/heatmap_pseudobulk.html">heatmap_pseudobulk</a></span><span class="op">(</span><span class="va">object</span>, </span>
<span>                     features <span class="op">=</span> <span class="va">dge</span><span class="op">$</span><span class="va">gene</span>, </span>
<span>                     groups <span class="op">=</span> <span class="st">"Patient"</span>,</span>
<span>                     idents <span class="op">=</span> <span class="va">temp_clusters</span>, </span>
<span>                     <span class="va">colors</span>, </span>
<span>                     file_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">temp_clusters</span>, <span class="st">"_heatmap.png"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS 26.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] future_1.68.0             yaml_2.3.11              </span></span>
<span><span class="co">#&gt;  [3] sessioninfo_1.2.3         matrixStats_1.5.0        </span></span>
<span><span class="co">#&gt;  [5] rlang_1.1.6               scCustomize_3.2.2        </span></span>
<span><span class="co">#&gt;  [7] convert2anndata_0.2.0     anndata_0.8.0            </span></span>
<span><span class="co">#&gt;  [9] pheatmap_1.0.13           ComplexHeatmap_2.25.2    </span></span>
<span><span class="co">#&gt; [11] harmony_1.2.4             Rcpp_1.1.0               </span></span>
<span><span class="co">#&gt; [13] Matrix_1.7-0              jsonlite_2.0.0           </span></span>
<span><span class="co">#&gt; [15] glue_1.8.0                stringr_1.6.0            </span></span>
<span><span class="co">#&gt; [17] treemap_2.4-4             RColorBrewer_1.1-3       </span></span>
<span><span class="co">#&gt; [19] eulerr_7.0.4              circlize_0.4.16          </span></span>
<span><span class="co">#&gt; [21] tibble_3.3.0              tidyr_1.3.1              </span></span>
<span><span class="co">#&gt; [23] dplyr_1.1.4               ggplot2_3.5.2            </span></span>
<span><span class="co">#&gt; [25] Seurat_5.1.0              SeuratObject_5.2.0       </span></span>
<span><span class="co">#&gt; [27] sp_2.2-0                  BioYourOwnBowl_0.0.0.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RcppAnnoy_0.0.22            splines_4.4.1              </span></span>
<span><span class="co">#&gt;   [3] later_1.4.4                 polyclip_1.10-7            </span></span>
<span><span class="co">#&gt;   [5] janitor_2.2.1               fastDummies_1.7.5          </span></span>
<span><span class="co">#&gt;   [7] lifecycle_1.0.4             rstatix_0.7.3              </span></span>
<span><span class="co">#&gt;   [9] doParallel_1.0.17           globals_0.18.0             </span></span>
<span><span class="co">#&gt;  [11] lattice_0.22-6              MASS_7.3-60.2              </span></span>
<span><span class="co">#&gt;  [13] backports_1.5.0             magrittr_2.0.4             </span></span>
<span><span class="co">#&gt;  [15] plotly_4.11.0               sass_0.4.10                </span></span>
<span><span class="co">#&gt;  [17] rmarkdown_2.30              jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [19] httpuv_1.6.16               otel_0.2.0                 </span></span>
<span><span class="co">#&gt;  [21] sctransform_0.4.2           spam_2.11-1                </span></span>
<span><span class="co">#&gt;  [23] spatstat.sparse_3.1-0       EnhancedVolcano_1.20.0     </span></span>
<span><span class="co">#&gt;  [25] reticulate_1.44.1           cowplot_1.2.0              </span></span>
<span><span class="co">#&gt;  [27] pbapply_1.7-4               lubridate_1.9.4            </span></span>
<span><span class="co">#&gt;  [29] abind_1.4-8                 zlibbioc_1.52.0            </span></span>
<span><span class="co">#&gt;  [31] Rtsne_0.17                  GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [33] purrr_1.2.0                 BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt;  [35] GenomeInfoDbData_1.2.13     IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [37] S4Vectors_0.44.0            ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt;  [39] irlba_2.3.5.1               listenv_0.10.0             </span></span>
<span><span class="co">#&gt;  [41] spatstat.utils_3.2-0        goftest_1.2-3              </span></span>
<span><span class="co">#&gt;  [43] RSpectra_0.16-2             spatstat.random_3.4-3      </span></span>
<span><span class="co">#&gt;  [45] fitdistrplus_1.2-4          parallelly_1.45.1          </span></span>
<span><span class="co">#&gt;  [47] pkgdown_2.2.0               leiden_0.4.3.1             </span></span>
<span><span class="co">#&gt;  [49] codetools_0.2-20            getopt_1.20.4              </span></span>
<span><span class="co">#&gt;  [51] DelayedArray_0.32.0         tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [53] shape_1.4.6.1               UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt;  [55] farver_2.1.2                stats4_4.4.1               </span></span>
<span><span class="co">#&gt;  [57] spatstat.explore_3.6-0      GetoptLong_1.1.0           </span></span>
<span><span class="co">#&gt;  [59] Formula_1.2-5               progressr_0.18.0           </span></span>
<span><span class="co">#&gt;  [61] ggridges_0.5.7              survival_3.6-4             </span></span>
<span><span class="co">#&gt;  [63] iterators_1.0.14            systemfonts_1.3.1          </span></span>
<span><span class="co">#&gt;  [65] foreach_1.5.2               tools_4.4.1                </span></span>
<span><span class="co">#&gt;  [67] ragg_1.5.0                  ica_1.0-3                  </span></span>
<span><span class="co">#&gt;  [69] gridExtra_2.3               SparseArray_1.6.2          </span></span>
<span><span class="co">#&gt;  [71] xfun_0.54                   MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt;  [73] GenomeInfoDb_1.42.3         withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [75] fastmap_1.2.0               digest_0.6.39              </span></span>
<span><span class="co">#&gt;  [77] timechange_0.3.0            R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [79] mime_0.13                   ggprism_1.0.7              </span></span>
<span><span class="co">#&gt;  [81] textshaping_1.0.4           colorspace_2.1-2           </span></span>
<span><span class="co">#&gt;  [83] Cairo_1.7-0                 scattermore_1.2            </span></span>
<span><span class="co">#&gt;  [85] tensor_1.5.1                spatstat.data_3.1-9        </span></span>
<span><span class="co">#&gt;  [87] generics_0.1.4              renv_1.1.0                 </span></span>
<span><span class="co">#&gt;  [89] data.table_1.17.8           httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [91] htmlwidgets_1.6.4           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt;  [93] uwot_0.2.4                  pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [95] gtable_0.3.6                lmtest_0.9-40              </span></span>
<span><span class="co">#&gt;  [97] SingleCellExperiment_1.28.1 XVector_0.46.0             </span></span>
<span><span class="co">#&gt;  [99] htmltools_0.5.9             carData_3.0-5              </span></span>
<span><span class="co">#&gt; [101] dotCall64_1.2               clue_0.3-66                </span></span>
<span><span class="co">#&gt; [103] scales_1.4.0                Biobase_2.66.0             </span></span>
<span><span class="co">#&gt; [105] png_0.1-8                   optparse_1.7.5             </span></span>
<span><span class="co">#&gt; [107] snakecase_0.11.1            spatstat.univar_3.1-5      </span></span>
<span><span class="co">#&gt; [109] knitr_1.50                  rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt; [111] reshape2_1.4.5              rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [113] nlme_3.1-164                cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [115] zoo_1.8-14                  GlobalOptions_0.1.3        </span></span>
<span><span class="co">#&gt; [117] KernSmooth_2.23-24          vipor_0.4.7                </span></span>
<span><span class="co">#&gt; [119] parallel_4.4.1              miniUI_0.1.2               </span></span>
<span><span class="co">#&gt; [121] ggrastr_1.0.2               desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [123] pillar_1.11.1               vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [125] RANN_2.6.2                  ggpubr_0.6.2               </span></span>
<span><span class="co">#&gt; [127] promises_1.5.0              car_3.1-3                  </span></span>
<span><span class="co">#&gt; [129] xtable_1.8-4                cluster_2.1.6              </span></span>
<span><span class="co">#&gt; [131] paletteer_1.6.0             beeswarm_0.4.0             </span></span>
<span><span class="co">#&gt; [133] evaluate_1.0.5              cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [135] compiler_4.4.1              crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [137] ggsignif_0.6.4              future.apply_1.20.0        </span></span>
<span><span class="co">#&gt; [139] labeling_0.4.3              rematch2_2.1.2             </span></span>
<span><span class="co">#&gt; [141] forcats_1.0.1               ggbeeswarm_0.7.3           </span></span>
<span><span class="co">#&gt; [143] plyr_1.8.9                  fs_1.6.6                   </span></span>
<span><span class="co">#&gt; [145] stringi_1.8.7               gridBase_0.4-7             </span></span>
<span><span class="co">#&gt; [147] viridisLite_0.4.2           deldir_2.0-4               </span></span>
<span><span class="co">#&gt; [149] assertthat_0.2.1            lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [151] spatstat.geom_3.6-1         RcppHNSW_0.6.0             </span></span>
<span><span class="co">#&gt; [153] patchwork_1.3.2             shiny_1.12.0               </span></span>
<span><span class="co">#&gt; [155] SummarizedExperiment_1.36.0 ROCR_1.0-11                </span></span>
<span><span class="co">#&gt; [157] broom_1.0.11                igraph_2.2.1               </span></span>
<span><span class="co">#&gt; [159] bslib_0.9.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Seurat: <a href="https://satijalab.org/seurat/" class="external-link uri">https://satijalab.org/seurat/</a>
</li>
<li>BioYourOwnBowl: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl" class="uri">https://github.com/june-zhang-bioinfo/BioYourOwnBowl</a>
</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by June Zhang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
