<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Consensus Non-negative Matrix Factorization (cNMF) Results Exploration • BioYourOwnBowl</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/journal/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Consensus Non-negative Matrix Factorization (cNMF) Results Exploration">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioYourOwnBowl</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioYourOwnBowl-cnmf.html">Consensus Non-negative Matrix Factorization (cNMF) Results Exploration</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-main.html">Getting Started with BioYourOwnBowl</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-vis.html">Customized Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Consensus Non-negative Matrix Factorization
(cNMF) Results Exploration</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/blob/HEAD/vignettes/BioYourOwnBowl-cnmf.Rmd"><code>vignettes/BioYourOwnBowl-cnmf.Rmd</code></a></small>
      <div class="hidden name"><code>BioYourOwnBowl-cnmf.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><strong>BioYourOwnBowl</strong> provides a comprehensive pipeline for
single-cell RNA-seq analysis using Seurat. This vignette demonstrates
the main workflow using example data from article: <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8376199/" class="external-link uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC8376199/</a>. The
seurat object is downsampled and has metadata included.</p>
<p>Consensus Non-negative Matrix Factorization is a widely used
computational method designed to identify gene expression programs
(GEPs) from single-cell RNA sequencing (scRNA-seq) data.</p>
<p>Given the fact that running cNMF and generating density plot requires
h5ad format, this vignette includes how to make h5ad format from seurat
object and create density plot.</p>
<p>This vignette provide comprehensive visualizations and data mining of
cNMF results, including: - Bar plots of top genes per program - Feature
plot on UMAP per program - Locate genes of interest per program (Are
your favorite genes in the programs? Where are they sitting?) - Assign
top programs per cell (Primary, Secondary, Tietary, etc) with cutoff -
Visualize Primary usage based on meta data - Compare the similarity of
cNMF (to self or siblings)</p>
<p>The results from this notebook can be found in <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/tree/main/vignettes/cnmf" class="uri">https://github.com/june-zhang-bioinfo/BioYourOwnBowl/tree/main/vignettes/cnmf</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"june-zhang-bioinfo/BioYourOwnBowl"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<pre><code><span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; 'SeuratObject' was built under R 4.4.0 but the current version is</span></span>
<span><span class="co">#&gt; 4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI</span></span>
<span><span class="co">#&gt; for R may have changed</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'SeuratObject'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; circlize version 0.4.16</span></span>
<span><span class="co">#&gt; CRAN page: https://cran.r-project.org/package=circlize</span></span>
<span><span class="co">#&gt; Github page: https://github.com/jokergoo/circlize</span></span>
<span><span class="co">#&gt; Documentation: https://jokergoo.github.io/circlize_book/book/</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span><span class="co">#&gt; Gu, Z. circlize implements and enhances circular visualization</span></span>
<span><span class="co">#&gt;   in R. Bioinformatics 2014.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; This message can be suppressed by:</span></span>
<span><span class="co">#&gt;   suppressPackageStartupMessages(library(circlize))</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Matrix'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:tidyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand, pack, unpack</span></span>
<span><span class="co">#&gt; Loading required package: Rcpp</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; ComplexHeatmap version 2.25.2</span></span>
<span><span class="co">#&gt; Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span><span class="co">#&gt; Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span><span class="co">#&gt; Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite either one:</span></span>
<span><span class="co">#&gt; - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span><span class="co">#&gt; - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span><span class="co">#&gt;     genomic data. Bioinformatics 2016.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The new InteractiveComplexHeatmap package can directly export static </span></span>
<span><span class="co">#&gt; complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; This message can be suppressed by:</span></span>
<span><span class="co">#&gt;   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'pheatmap'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ComplexHeatmap':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     pheatmap</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'anndata'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:SeuratObject':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Layers</span></span>
<span><span class="co">#&gt; scCustomize v3.2.2</span></span>
<span><span class="co">#&gt; If you find the scCustomize useful please cite.</span></span>
<span><span class="co">#&gt; See 'samuel-marsh.github.io/scCustomize/articles/FAQ.html' for citation info.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'rlang'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:jsonlite':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     flatten, unbox</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'matrixStats'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     count</span></span>
<span><span class="co">#&gt; Warning: package 'yaml' was built under R version 4.4.3</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="prepare-example-data">Prepare Example Data<a class="anchor" aria-label="anchor" href="#prepare-example-data"></a>
</h2>
<p>Load the processed object from the main pipeline.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_dir_base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"BioYourOwnBowl"</span><span class="op">)</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"GSE178756_processed.rds"</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">data_path</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 9453 features across 5893 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (9453 features, 2000 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  2 dimensional reductions calculated: pca, umap</span></span>
<span></span>
<span><span class="va">cnmf_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"cnmf_k12"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-color-palette">Define Color Palette<a class="anchor" aria-label="anchor" href="#define-color-palette"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-selected-features">Define Selected Features<a class="anchor" aria-label="anchor" href="#define-selected-features"></a>
</h2>
<p>You can provide specific genes of interest to highlight in
visualizations:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"CD8A"</span>, <span class="st">"CD8B"</span>,</span>
<span>  <span class="st">"CD3D"</span>, <span class="st">"CD3E"</span>, <span class="st">"CD3G"</span>,</span>
<span>  <span class="st">"GZMB"</span>, <span class="st">"GZMA"</span>, <span class="st">"GZMH"</span>,</span>
<span>  <span class="st">"PRF1"</span>, <span class="st">"NKG7"</span>, <span class="st">"GNLY"</span>,</span>
<span>  <span class="st">"IL2RA"</span>, <span class="st">"TNFRSF9"</span>, <span class="st">"CD69"</span>,</span>
<span>  <span class="st">"MKI67"</span>, <span class="st">"TOP2A"</span>,</span>
<span>  <span class="st">"PDCD1"</span>, <span class="st">"LAG3"</span>, <span class="st">"TIGIT"</span>,</span>
<span>  <span class="st">"CTLA4"</span>, <span class="st">"HAVCR2"</span>,</span>
<span>  <span class="st">"CCR7"</span>, <span class="st">"SELL"</span>, <span class="st">"TCF7"</span>,</span>
<span>  <span class="st">"LEF1"</span>, <span class="st">"IL7R"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="prepare-h5ad-file-to-run-cnmf-and-density-plot-scanpy">Prepare h5ad file to run cNMF and density plot (scanpy)<a class="anchor" aria-label="anchor" href="#prepare-h5ad-file-to-run-cnmf-and-density-plot-scanpy"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># to use reticulate, you need to initialize your virtual environment</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span><span class="fu">BioYourOwnBowl</span><span class="fu">::</span><span class="fu"><a href="../reference/setup_python.html">setup_python</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepare_h5ad.html">prepare_h5ad</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">seurat_obj</span>, features <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, metadata_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cohort"</span>, <span class="st">"Time_point"</span><span class="op">)</span>, file_name <span class="op">=</span> <span class="st">"example.h5ad"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">py_code</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/density_plot.html">density_plot</a></span><span class="op">(</span>file_name <span class="op">=</span> <span class="st">"example.h5ad"</span>, density_specs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>   Cohort <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HCV"</span>, <span class="st">"Healthy"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>   Time_point <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Post_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-top-genes-per-program">Visualize top genes per program<a class="anchor" aria-label="anchor" href="#visualize-top-genes-per-program"></a>
</h3>
<p><code>highlight_genes</code> can be turned off.
<code>show_metrics</code> can be turned on if you are interested in the
median expression of genes.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cnmf_bar_plots.html">cnmf_bar_plots</a></span><span class="op">(</span></span>
<span>         <span class="va">cnmf_path</span>,</span>
<span>         <span class="va">seurat_obj</span>,</span>
<span>         top_n <span class="op">=</span> <span class="fl">30</span>,</span>
<span>         highlight_genes <span class="op">=</span> <span class="va">selected_features</span>,</span>
<span>         output_dir <span class="op">=</span> <span class="st">"/Users/zz005/Documents/github/R/vignettes3_results"</span>,</span>
<span>         show_metrics <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="co">#&gt; Finished generating plots in: /Users/zz005/Documents/github/R/vignettes3_results</span></span>
<span><span class="va">r</span><span class="op">$</span><span class="va">top_df</span></span>
<span><span class="co">#&gt;    Usage_1  Usage_2 Usage_3  Usage_4   Usage_5 Usage_6   Usage_7  Usage_8</span></span>
<span><span class="co">#&gt; 1     GNLY     CCL5  S100A9   EEF1A1   HLA.DRA   KLRB1       JUN   EEF1A1</span></span>
<span><span class="co">#&gt; 2     GZMH     TPT1     LYZ     TPT1      CD74    CCL5    DNAJB1     TPT1</span></span>
<span><span class="co">#&gt; 3     GZMB     NKG7  S100A8     ACTB    EEF1A1  EEF1A1       FOS      LTB</span></span>
<span><span class="co">#&gt; 4   FGFBP2   EEF1A1     FTL     KLF2      TPT1    TPT1      JUND     BTG1</span></span>
<span><span class="co">#&gt; 5     NKG7     GZMA    VCAN     LDHB  HLA.DRB1 ZFP36L2     DUSP1      FTL</span></span>
<span><span class="co">#&gt; 6     PRF1     BTG1    FTH1     BTG1  HLA.DPA1   DUSP2      JUNB     ACTB</span></span>
<span><span class="co">#&gt; 7     CCL5     LCP1    CTSS      LTB  HLA.DQB1   CXCR4    HSPA1A     IL7R</span></span>
<span><span class="co">#&gt; 8     ACTB  ZFP36L2    TPT1      FTL  HLA.DPB1    IL7R    HSPA1B    TENM1</span></span>
<span><span class="co">#&gt; 9   S100A4     GZMK    ACTB    H3F3B     MS4A1    NKG7   TSC22D3     CCL5</span></span>
<span><span class="co">#&gt; 10  EEF1A1     KLF2  EEF1A1    ACTG1       FTL    GZMA      CD69   S100A4</span></span>
<span><span class="co">#&gt; 11    TPT1 HSP90AA1    FCN1   PABPC1       LTB    GZMK     H3F3B    S100B</span></span>
<span><span class="co">#&gt; 12   EFHD2     IL7R     VIM    EIF3E     CD79A   ARL4C  HSP90AA1     CUBN</span></span>
<span><span class="co">#&gt; 13    CST7     JUND     FOS     JUNB      ACTB    BTG1      KLF6    MYOM2</span></span>
<span><span class="co">#&gt; 14   KLRD1    SYNE2  S100A4  ZFP36L2  HLA.DQA1    ACTB  PPP1R15A  PLA2G4B</span></span>
<span><span class="co">#&gt; 15   ACTG1    ARL4C  S100A6     FTH1      FTH1    LYAR     ZFP36      AK1</span></span>
<span><span class="co">#&gt; 16    PFN1     ACTB    CST3     JUND      BTG1  S100A4     DUSP2      VIM</span></span>
<span><span class="co">#&gt; 17  LGALS1    AHNAK    MNDA    NOSIP     ACTG1    JUND      KLF2    LMOD3</span></span>
<span><span class="co">#&gt; 18    GZMA    H3F3B   GAPDH     TRAC     CD79B    JUNB      IER2    AKAP5</span></span>
<span><span class="co">#&gt; 19    CTSW      FTL  TYROBP    FOXP1     H3F3B   H3F3B MTRNR2L12    SGPP2</span></span>
<span><span class="co">#&gt; 20  S100A6     CST7    SRGN     TCF7   HLA.DMA   KLRG1      BTG2    ISG15</span></span>
<span><span class="co">#&gt; 21     FTL      JUN HLA.DRA     SELL      KLF2     FTL     CXCR4 TNFRSF18</span></span>
<span><span class="co">#&gt; 22   CRIP1   PIK3R1    PSAP LEPROTL1    PABPC1    CTSW      BTG1  TNFRSF4</span></span>
<span><span class="co">#&gt; 23  FCGR3A      VIM S100A12     PFN1      JUND    KLF2     HSPE1     NADK</span></span>
<span><span class="co">#&gt; 24   TRGC2    SYNE1  PABPC1    TRBC2   RALGPS2    TRDC    EEF1A1 TNFRSF25</span></span>
<span><span class="co">#&gt; 25    HOPX     JUNB S100A11     IL7R       VIM    FTH1      SAT1     PER3</span></span>
<span><span class="co">#&gt; 26   GAPDH    CXCR4    SAT1    CXCR4     CRIP1    CST7    NFKBIA SLC25A33</span></span>
<span><span class="co">#&gt; 27    ZEB2     CMC1    CD14      VIM     EIF3E    CD69    SLC2A3     RBP7</span></span>
<span><span class="co">#&gt; 28   H3F3B    ITGA4    AIF1    RGS10  HSP90AA1   TRGC1      TPT1    KIF1B</span></span>
<span><span class="co">#&gt; 29    SRGN   S100A4   ACTG1      CD7 TNFRSF13C   ACTG1   ZFP36L2      PGD</span></span>
<span><span class="co">#&gt; 30   SPON2     KLF6    TYMP     FYB1      UCP2   DUSP1   HNRNPH1   AGTRAP</span></span>
<span><span class="co">#&gt;    Usage_9 Usage_10 Usage_11 Usage_12</span></span>
<span><span class="co">#&gt; 1   S100A4     CCL5     TPT1      FTL</span></span>
<span><span class="co">#&gt; 2      LTB     NKG7   EEF1A1     FTH1</span></span>
<span><span class="co">#&gt; 3   EEF1A1     ACTB    SYNE2     ACTB</span></span>
<span><span class="co">#&gt; 4    CRIP1     PFN1     IL7R   EEF1A1</span></span>
<span><span class="co">#&gt; 5     ACTB   EEF1A1    AHNAK     TPT1</span></span>
<span><span class="co">#&gt; 6      VIM     CMC1     BTG1     AIF1</span></span>
<span><span class="co">#&gt; 7     TPT1     TRAC     FYB1     CST3</span></span>
<span><span class="co">#&gt; 8   S100A6      FTL     UTRN   S100A4</span></span>
<span><span class="co">#&gt; 9     FTH1     CTSW    H3F3B    COTL1</span></span>
<span><span class="co">#&gt; 10   ACTG1     GZMA   VPS13C     LST1</span></span>
<span><span class="co">#&gt; 11     FTL   S100A4    ITGB1  HLA.DRA</span></span>
<span><span class="co">#&gt; 12    TRAC     FTH1   SAMHD1 HLA.DPA1</span></span>
<span><span class="co">#&gt; 13 S100A11     CD8B     BCL2     CTSS</span></span>
<span><span class="co">#&gt; 14 S100A10     CST7     PAG1   S100A6</span></span>
<span><span class="co">#&gt; 15    PFN1   S100A6      VIM     SAT1</span></span>
<span><span class="co">#&gt; 16    LDHB    GAPDH      LTB     PSAP</span></span>
<span><span class="co">#&gt; 17    IL7R     CD74  HNRNPH1     CD74</span></span>
<span><span class="co">#&gt; 18   GAPDH    ACTG1   CDC14A    IFI30</span></span>
<span><span class="co">#&gt; 19   ANXA1    H3F3B   PABPC1   FCER1G</span></span>
<span><span class="co">#&gt; 20   FXYD5     TPT1    FOXP1   TYROBP</span></span>
<span><span class="co">#&gt; 21   H3F3B HLA.DPB1     ACTB   FCGR3A</span></span>
<span><span class="co">#&gt; 22   EIF3E     GZMH    SORL1     TYMP</span></span>
<span><span class="co">#&gt; 23    BTG1    TRBC2   INPP4B    ACTG1</span></span>
<span><span class="co">#&gt; 24   TRBC2    KLRK1    SLFN5  S100A11</span></span>
<span><span class="co">#&gt; 25    PLP2     CD8A     KLF2 HLA.DPB1</span></span>
<span><span class="co">#&gt; 26  PABPC1  S100A10     KLF6      LYZ</span></span>
<span><span class="co">#&gt; 27  LGALS1 HLA.DPA1 HIST1H1E      VIM</span></span>
<span><span class="co">#&gt; 28   ITGB1    CRIP1  ANKRD11 HLA.DRB1</span></span>
<span><span class="co">#&gt; 29    JUNB    KLRD1   ATF7IP      CFD</span></span>
<span><span class="co">#&gt; 30    TSPO     GZMK  LRRFIP1    H3F3B</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-ranks-of-genes-of-interest">Get ranks of genes of interest<a class="anchor" aria-label="anchor" href="#get-ranks-of-genes-of-interest"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cnmf_gene_ranks.html">cnmf_gene_ranks</a></span><span class="op">(</span></span>
<span>    dir_path <span class="op">=</span> <span class="va">cnmf_path</span>,</span>
<span>    features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TOX"</span>, <span class="st">"PDCD1"</span>, <span class="st">"SELL"</span>, <span class="st">"IKZF2"</span>, <span class="st">"BACH2"</span>, <span class="st">"ENTPD1"</span>, <span class="st">"ETV1"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in cnmf_gene_ranks(dir_path = cnmf_path, features = c("TOX", "PDCD1", :</span></span>
<span><span class="co">#&gt; The following genes were not found in cNMF results: PDCD1, ETV1</span></span>
<span><span class="co">#&gt; Total genes in analysis: 2000 </span></span>
<span><span class="co">#&gt; Total programs in analysis: 12 </span></span>
<span><span class="co">#&gt; ======================</span></span>
<span><span class="co">#&gt;        Usage_1 Usage_2 Usage_3 Usage_4 Usage_5 Usage_6 Usage_7 Usage_8 Usage_9</span></span>
<span><span class="co">#&gt; TOX        368     700    1743    1320     905    1567     743      20    1583</span></span>
<span><span class="co">#&gt; SELL      1458     393     128      21      76     138     755      20     164</span></span>
<span><span class="co">#&gt; IKZF2      362    1222    1743    1471     970     705    1245      20    1583</span></span>
<span><span class="co">#&gt; BACH2     1264    1363    1743     381     504    1188     980      20    1539</span></span>
<span><span class="co">#&gt; ENTPD1    1458    1670     390    1548     211    1567    1306      20    1500</span></span>
<span><span class="co">#&gt;        Usage_10 Usage_11 Usage_12</span></span>
<span><span class="co">#&gt; TOX         537     1063     1634</span></span>
<span><span class="co">#&gt; SELL        319      262     1634</span></span>
<span><span class="co">#&gt; IKZF2       842      393     1600</span></span>
<span><span class="co">#&gt; BACH2       842      646     1634</span></span>
<span><span class="co">#&gt; ENTPD1      842      985      851</span></span>
<span><span class="va">ranks</span></span>
<span><span class="co">#&gt;        Usage_1 Usage_2 Usage_3 Usage_4 Usage_5 Usage_6 Usage_7 Usage_8 Usage_9</span></span>
<span><span class="co">#&gt; TOX        368     700    1743    1320     905    1567     743      20    1583</span></span>
<span><span class="co">#&gt; SELL      1458     393     128      21      76     138     755      20     164</span></span>
<span><span class="co">#&gt; IKZF2      362    1222    1743    1471     970     705    1245      20    1583</span></span>
<span><span class="co">#&gt; BACH2     1264    1363    1743     381     504    1188     980      20    1539</span></span>
<span><span class="co">#&gt; ENTPD1    1458    1670     390    1548     211    1567    1306      20    1500</span></span>
<span><span class="co">#&gt;        Usage_10 Usage_11 Usage_12</span></span>
<span><span class="co">#&gt; TOX         537     1063     1634</span></span>
<span><span class="co">#&gt; SELL        319      262     1634</span></span>
<span><span class="co">#&gt; IKZF2       842      393     1600</span></span>
<span><span class="co">#&gt; BACH2       842      646     1634</span></span>
<span><span class="co">#&gt; ENTPD1      842      985      851</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="project-cnmf-programs-on-umap">Project cNMF programs on UMAP<a class="anchor" aria-label="anchor" href="#project-cnmf-programs-on-umap"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/cnmf_umaps.html">cnmf_umaps</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">seurat_obj</span>, </span>
<span>           dir_path <span class="op">=</span> <span class="va">cnmf_path</span>, </span>
<span>           output_dir <span class="op">=</span> <span class="st">"/Users/zz005/Documents/github/R/vignettes3_results"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `FetchData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; NOTE: <span style="color: #00BB00;">FeaturePlot_scCustom</span> uses a specified `na_cutoff` when plotting to</span></span>
<span><span class="co">#&gt; color cells with no expression as background color separate from color scale.</span></span>
<span><span class="co">#&gt; Please ensure `na_cutoff` value is appropriate for feature being plotted.</span></span>
<span><span class="co">#&gt; Default setting is appropriate for use when plotting from 'RNA' assay.</span></span>
<span><span class="co">#&gt; When `na_cutoff` not appropriate (e.g., module scores) set to NULL to</span></span>
<span><span class="co">#&gt; plot all cells in gradient color palette.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; -----This message will be shown once per session.-----</span></span>
<span><span class="co">#&gt; Warning: Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Warning: Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; Some of the plotted features are from meta.data slot.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> Please check that `na_cutoff` param is being set appropriately for those</span></span>
<span><span class="co">#&gt;   features.</span></span>
<span><span class="co">#&gt; UMAP usage plots saved to: /Users/zz005/Documents/github/R/vignettes3_results</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 9453 features across 5893 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (9453 features, 2000 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  2 dimensional reductions calculated: pca, umap</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="assign-top-programs-to-cells-primary-secondary-tietiary-quaternary">Assign top programs to cells (Primary, Secondary, Tietiary,
Quaternary)<a class="anchor" aria-label="anchor" href="#assign-top-programs-to-cells-primary-secondary-tietiary-quaternary"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cnmf_top_programs.html">cnmf_top_programs</a></span><span class="op">(</span><span class="va">seurat_obj</span>, <span class="va">cnmf_path</span>, cutoff <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="va">r</span><span class="op">$</span><span class="va">object</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">r</span><span class="op">$</span><span class="va">cell_by_pro</span> <span class="co"># used for sankey plot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare_cnmf_programs">compare_cnmf_programs<a class="anchor" aria-label="anchor" href="#compare_cnmf_programs"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compare_cnmf_programs.html">compare_cnmf_programs</a></span><span class="op">(</span></span>
<span>    dir1 <span class="op">=</span> <span class="va">cnmf_path</span>,</span>
<span>    dir2 <span class="op">=</span> <span class="va">cnmf_path</span>,</span>
<span>    object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>    top_n <span class="op">=</span> <span class="fl">30</span>,</span>
<span>    label1 <span class="op">=</span> <span class="st">"k12"</span>,</span>
<span>    label2 <span class="op">=</span> <span class="st">"k12"</span>,</span>
<span>    heatmap_title <span class="op">=</span> <span class="st">""</span>,</span>
<span>    vf_only <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-cnmf_files/figure-html/unnamed-chunk-13-1.png" class="r-plt" width="400"></p>
<pre><code><span><span class="co">#&gt; $overlap_matrix</span></span>
<span><span class="co">#&gt;        k12_1 k12_2 k12_3 k12_4 k12_5 k12_6 k12_7 k12_8 k12_9 k12_10 k12_11</span></span>
<span><span class="co">#&gt; k12_1     30    10     9     7     7    12     3     6    12     18      4</span></span>
<span><span class="co">#&gt; k12_2     10    30     6    13    10    19    12     9    10     12     11</span></span>
<span><span class="co">#&gt; k12_3      9     6    30     8     9     7     4     6    12      9      5</span></span>
<span><span class="co">#&gt; k12_4      7    13     8    30    14    14     9     8    18     10     12</span></span>
<span><span class="co">#&gt; k12_5      7    10     9    14    30    10     7     7    13     11      9</span></span>
<span><span class="co">#&gt; k12_6     12    19     7    14    10    30    12     8    11     14      7</span></span>
<span><span class="co">#&gt; k12_7      3    12     4     9     7    12    30     3     5      3      7</span></span>
<span><span class="co">#&gt; k12_8      6     9     6     8     7     8     3    30     9      6      7</span></span>
<span><span class="co">#&gt; k12_9     12    10    12    18    13    11     5     9    30     15     10</span></span>
<span><span class="co">#&gt; k12_10    18    12     9    10    11    14     3     6    15     30      4</span></span>
<span><span class="co">#&gt; k12_11     4    11     5    12     9     7     7     7    10      4     30</span></span>
<span><span class="co">#&gt; k12_12     9     7    19     8    13     8     4     6    11     12      5</span></span>
<span><span class="co">#&gt;        k12_12</span></span>
<span><span class="co">#&gt; k12_1       9</span></span>
<span><span class="co">#&gt; k12_2       7</span></span>
<span><span class="co">#&gt; k12_3      19</span></span>
<span><span class="co">#&gt; k12_4       8</span></span>
<span><span class="co">#&gt; k12_5      13</span></span>
<span><span class="co">#&gt; k12_6       8</span></span>
<span><span class="co">#&gt; k12_7       4</span></span>
<span><span class="co">#&gt; k12_8       6</span></span>
<span><span class="co">#&gt; k12_9      11</span></span>
<span><span class="co">#&gt; k12_10     12</span></span>
<span><span class="co">#&gt; k12_11      5</span></span>
<span><span class="co">#&gt; k12_12     30</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $top_df1</span></span>
<span><span class="co">#&gt;    Usage_1  Usage_2 Usage_3  Usage_4   Usage_5 Usage_6   Usage_7  Usage_8</span></span>
<span><span class="co">#&gt; 1     GNLY     CCL5  S100A9   EEF1A1   HLA.DRA   KLRB1       JUN   EEF1A1</span></span>
<span><span class="co">#&gt; 2     GZMH     TPT1     LYZ     TPT1      CD74    CCL5    DNAJB1     TPT1</span></span>
<span><span class="co">#&gt; 3     GZMB     NKG7  S100A8     ACTB    EEF1A1  EEF1A1       FOS      LTB</span></span>
<span><span class="co">#&gt; 4   FGFBP2   EEF1A1     FTL     KLF2      TPT1    TPT1      JUND     BTG1</span></span>
<span><span class="co">#&gt; 5     NKG7     GZMA    VCAN     LDHB  HLA.DRB1 ZFP36L2     DUSP1      FTL</span></span>
<span><span class="co">#&gt; 6     PRF1     BTG1    FTH1     BTG1  HLA.DPA1   DUSP2      JUNB     ACTB</span></span>
<span><span class="co">#&gt; 7     CCL5     LCP1    CTSS      LTB  HLA.DQB1   CXCR4    HSPA1A     IL7R</span></span>
<span><span class="co">#&gt; 8     ACTB  ZFP36L2    TPT1      FTL  HLA.DPB1    IL7R    HSPA1B    TENM1</span></span>
<span><span class="co">#&gt; 9   S100A4     GZMK    ACTB    H3F3B     MS4A1    NKG7   TSC22D3     CCL5</span></span>
<span><span class="co">#&gt; 10  EEF1A1     KLF2  EEF1A1    ACTG1       FTL    GZMA      CD69   S100A4</span></span>
<span><span class="co">#&gt; 11    TPT1 HSP90AA1    FCN1   PABPC1       LTB    GZMK     H3F3B    S100B</span></span>
<span><span class="co">#&gt; 12   EFHD2     IL7R     VIM    EIF3E     CD79A   ARL4C  HSP90AA1     CUBN</span></span>
<span><span class="co">#&gt; 13    CST7     JUND     FOS     JUNB      ACTB    BTG1      KLF6    MYOM2</span></span>
<span><span class="co">#&gt; 14   KLRD1    SYNE2  S100A4  ZFP36L2  HLA.DQA1    ACTB  PPP1R15A  PLA2G4B</span></span>
<span><span class="co">#&gt; 15   ACTG1    ARL4C  S100A6     FTH1      FTH1    LYAR     ZFP36      AK1</span></span>
<span><span class="co">#&gt; 16    PFN1     ACTB    CST3     JUND      BTG1  S100A4     DUSP2      VIM</span></span>
<span><span class="co">#&gt; 17  LGALS1    AHNAK    MNDA    NOSIP     ACTG1    JUND      KLF2    LMOD3</span></span>
<span><span class="co">#&gt; 18    GZMA    H3F3B   GAPDH     TRAC     CD79B    JUNB      IER2    AKAP5</span></span>
<span><span class="co">#&gt; 19    CTSW      FTL  TYROBP    FOXP1     H3F3B   H3F3B MTRNR2L12    SGPP2</span></span>
<span><span class="co">#&gt; 20  S100A6     CST7    SRGN     TCF7   HLA.DMA   KLRG1      BTG2    ISG15</span></span>
<span><span class="co">#&gt; 21     FTL      JUN HLA.DRA     SELL      KLF2     FTL     CXCR4 TNFRSF18</span></span>
<span><span class="co">#&gt; 22   CRIP1   PIK3R1    PSAP LEPROTL1    PABPC1    CTSW      BTG1  TNFRSF4</span></span>
<span><span class="co">#&gt; 23  FCGR3A      VIM S100A12     PFN1      JUND    KLF2     HSPE1     NADK</span></span>
<span><span class="co">#&gt; 24   TRGC2    SYNE1  PABPC1    TRBC2   RALGPS2    TRDC    EEF1A1 TNFRSF25</span></span>
<span><span class="co">#&gt; 25    HOPX     JUNB S100A11     IL7R       VIM    FTH1      SAT1     PER3</span></span>
<span><span class="co">#&gt; 26   GAPDH    CXCR4    SAT1    CXCR4     CRIP1    CST7    NFKBIA SLC25A33</span></span>
<span><span class="co">#&gt; 27    ZEB2     CMC1    CD14      VIM     EIF3E    CD69    SLC2A3     RBP7</span></span>
<span><span class="co">#&gt; 28   H3F3B    ITGA4    AIF1    RGS10  HSP90AA1   TRGC1      TPT1    KIF1B</span></span>
<span><span class="co">#&gt; 29    SRGN   S100A4   ACTG1      CD7 TNFRSF13C   ACTG1   ZFP36L2      PGD</span></span>
<span><span class="co">#&gt; 30   SPON2     KLF6    TYMP     FYB1      UCP2   DUSP1   HNRNPH1   AGTRAP</span></span>
<span><span class="co">#&gt;    Usage_9 Usage_10 Usage_11 Usage_12</span></span>
<span><span class="co">#&gt; 1   S100A4     CCL5     TPT1      FTL</span></span>
<span><span class="co">#&gt; 2      LTB     NKG7   EEF1A1     FTH1</span></span>
<span><span class="co">#&gt; 3   EEF1A1     ACTB    SYNE2     ACTB</span></span>
<span><span class="co">#&gt; 4    CRIP1     PFN1     IL7R   EEF1A1</span></span>
<span><span class="co">#&gt; 5     ACTB   EEF1A1    AHNAK     TPT1</span></span>
<span><span class="co">#&gt; 6      VIM     CMC1     BTG1     AIF1</span></span>
<span><span class="co">#&gt; 7     TPT1     TRAC     FYB1     CST3</span></span>
<span><span class="co">#&gt; 8   S100A6      FTL     UTRN   S100A4</span></span>
<span><span class="co">#&gt; 9     FTH1     CTSW    H3F3B    COTL1</span></span>
<span><span class="co">#&gt; 10   ACTG1     GZMA   VPS13C     LST1</span></span>
<span><span class="co">#&gt; 11     FTL   S100A4    ITGB1  HLA.DRA</span></span>
<span><span class="co">#&gt; 12    TRAC     FTH1   SAMHD1 HLA.DPA1</span></span>
<span><span class="co">#&gt; 13 S100A11     CD8B     BCL2     CTSS</span></span>
<span><span class="co">#&gt; 14 S100A10     CST7     PAG1   S100A6</span></span>
<span><span class="co">#&gt; 15    PFN1   S100A6      VIM     SAT1</span></span>
<span><span class="co">#&gt; 16    LDHB    GAPDH      LTB     PSAP</span></span>
<span><span class="co">#&gt; 17    IL7R     CD74  HNRNPH1     CD74</span></span>
<span><span class="co">#&gt; 18   GAPDH    ACTG1   CDC14A    IFI30</span></span>
<span><span class="co">#&gt; 19   ANXA1    H3F3B   PABPC1   FCER1G</span></span>
<span><span class="co">#&gt; 20   FXYD5     TPT1    FOXP1   TYROBP</span></span>
<span><span class="co">#&gt; 21   H3F3B HLA.DPB1     ACTB   FCGR3A</span></span>
<span><span class="co">#&gt; 22   EIF3E     GZMH    SORL1     TYMP</span></span>
<span><span class="co">#&gt; 23    BTG1    TRBC2   INPP4B    ACTG1</span></span>
<span><span class="co">#&gt; 24   TRBC2    KLRK1    SLFN5  S100A11</span></span>
<span><span class="co">#&gt; 25    PLP2     CD8A     KLF2 HLA.DPB1</span></span>
<span><span class="co">#&gt; 26  PABPC1  S100A10     KLF6      LYZ</span></span>
<span><span class="co">#&gt; 27  LGALS1 HLA.DPA1 HIST1H1E      VIM</span></span>
<span><span class="co">#&gt; 28   ITGB1    CRIP1  ANKRD11 HLA.DRB1</span></span>
<span><span class="co">#&gt; 29    JUNB    KLRD1   ATF7IP      CFD</span></span>
<span><span class="co">#&gt; 30    TSPO     GZMK  LRRFIP1    H3F3B</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $top_df2</span></span>
<span><span class="co">#&gt;    Usage_1  Usage_2 Usage_3  Usage_4   Usage_5 Usage_6   Usage_7  Usage_8</span></span>
<span><span class="co">#&gt; 1     GNLY     CCL5  S100A9   EEF1A1   HLA.DRA   KLRB1       JUN   EEF1A1</span></span>
<span><span class="co">#&gt; 2     GZMH     TPT1     LYZ     TPT1      CD74    CCL5    DNAJB1     TPT1</span></span>
<span><span class="co">#&gt; 3     GZMB     NKG7  S100A8     ACTB    EEF1A1  EEF1A1       FOS      LTB</span></span>
<span><span class="co">#&gt; 4   FGFBP2   EEF1A1     FTL     KLF2      TPT1    TPT1      JUND     BTG1</span></span>
<span><span class="co">#&gt; 5     NKG7     GZMA    VCAN     LDHB  HLA.DRB1 ZFP36L2     DUSP1      FTL</span></span>
<span><span class="co">#&gt; 6     PRF1     BTG1    FTH1     BTG1  HLA.DPA1   DUSP2      JUNB     ACTB</span></span>
<span><span class="co">#&gt; 7     CCL5     LCP1    CTSS      LTB  HLA.DQB1   CXCR4    HSPA1A     IL7R</span></span>
<span><span class="co">#&gt; 8     ACTB  ZFP36L2    TPT1      FTL  HLA.DPB1    IL7R    HSPA1B    TENM1</span></span>
<span><span class="co">#&gt; 9   S100A4     GZMK    ACTB    H3F3B     MS4A1    NKG7   TSC22D3     CCL5</span></span>
<span><span class="co">#&gt; 10  EEF1A1     KLF2  EEF1A1    ACTG1       FTL    GZMA      CD69   S100A4</span></span>
<span><span class="co">#&gt; 11    TPT1 HSP90AA1    FCN1   PABPC1       LTB    GZMK     H3F3B    S100B</span></span>
<span><span class="co">#&gt; 12   EFHD2     IL7R     VIM    EIF3E     CD79A   ARL4C  HSP90AA1     CUBN</span></span>
<span><span class="co">#&gt; 13    CST7     JUND     FOS     JUNB      ACTB    BTG1      KLF6    MYOM2</span></span>
<span><span class="co">#&gt; 14   KLRD1    SYNE2  S100A4  ZFP36L2  HLA.DQA1    ACTB  PPP1R15A  PLA2G4B</span></span>
<span><span class="co">#&gt; 15   ACTG1    ARL4C  S100A6     FTH1      FTH1    LYAR     ZFP36      AK1</span></span>
<span><span class="co">#&gt; 16    PFN1     ACTB    CST3     JUND      BTG1  S100A4     DUSP2      VIM</span></span>
<span><span class="co">#&gt; 17  LGALS1    AHNAK    MNDA    NOSIP     ACTG1    JUND      KLF2    LMOD3</span></span>
<span><span class="co">#&gt; 18    GZMA    H3F3B   GAPDH     TRAC     CD79B    JUNB      IER2    AKAP5</span></span>
<span><span class="co">#&gt; 19    CTSW      FTL  TYROBP    FOXP1     H3F3B   H3F3B MTRNR2L12    SGPP2</span></span>
<span><span class="co">#&gt; 20  S100A6     CST7    SRGN     TCF7   HLA.DMA   KLRG1      BTG2    ISG15</span></span>
<span><span class="co">#&gt; 21     FTL      JUN HLA.DRA     SELL      KLF2     FTL     CXCR4 TNFRSF18</span></span>
<span><span class="co">#&gt; 22   CRIP1   PIK3R1    PSAP LEPROTL1    PABPC1    CTSW      BTG1  TNFRSF4</span></span>
<span><span class="co">#&gt; 23  FCGR3A      VIM S100A12     PFN1      JUND    KLF2     HSPE1     NADK</span></span>
<span><span class="co">#&gt; 24   TRGC2    SYNE1  PABPC1    TRBC2   RALGPS2    TRDC    EEF1A1 TNFRSF25</span></span>
<span><span class="co">#&gt; 25    HOPX     JUNB S100A11     IL7R       VIM    FTH1      SAT1     PER3</span></span>
<span><span class="co">#&gt; 26   GAPDH    CXCR4    SAT1    CXCR4     CRIP1    CST7    NFKBIA SLC25A33</span></span>
<span><span class="co">#&gt; 27    ZEB2     CMC1    CD14      VIM     EIF3E    CD69    SLC2A3     RBP7</span></span>
<span><span class="co">#&gt; 28   H3F3B    ITGA4    AIF1    RGS10  HSP90AA1   TRGC1      TPT1    KIF1B</span></span>
<span><span class="co">#&gt; 29    SRGN   S100A4   ACTG1      CD7 TNFRSF13C   ACTG1   ZFP36L2      PGD</span></span>
<span><span class="co">#&gt; 30   SPON2     KLF6    TYMP     FYB1      UCP2   DUSP1   HNRNPH1   AGTRAP</span></span>
<span><span class="co">#&gt;    Usage_9 Usage_10 Usage_11 Usage_12</span></span>
<span><span class="co">#&gt; 1   S100A4     CCL5     TPT1      FTL</span></span>
<span><span class="co">#&gt; 2      LTB     NKG7   EEF1A1     FTH1</span></span>
<span><span class="co">#&gt; 3   EEF1A1     ACTB    SYNE2     ACTB</span></span>
<span><span class="co">#&gt; 4    CRIP1     PFN1     IL7R   EEF1A1</span></span>
<span><span class="co">#&gt; 5     ACTB   EEF1A1    AHNAK     TPT1</span></span>
<span><span class="co">#&gt; 6      VIM     CMC1     BTG1     AIF1</span></span>
<span><span class="co">#&gt; 7     TPT1     TRAC     FYB1     CST3</span></span>
<span><span class="co">#&gt; 8   S100A6      FTL     UTRN   S100A4</span></span>
<span><span class="co">#&gt; 9     FTH1     CTSW    H3F3B    COTL1</span></span>
<span><span class="co">#&gt; 10   ACTG1     GZMA   VPS13C     LST1</span></span>
<span><span class="co">#&gt; 11     FTL   S100A4    ITGB1  HLA.DRA</span></span>
<span><span class="co">#&gt; 12    TRAC     FTH1   SAMHD1 HLA.DPA1</span></span>
<span><span class="co">#&gt; 13 S100A11     CD8B     BCL2     CTSS</span></span>
<span><span class="co">#&gt; 14 S100A10     CST7     PAG1   S100A6</span></span>
<span><span class="co">#&gt; 15    PFN1   S100A6      VIM     SAT1</span></span>
<span><span class="co">#&gt; 16    LDHB    GAPDH      LTB     PSAP</span></span>
<span><span class="co">#&gt; 17    IL7R     CD74  HNRNPH1     CD74</span></span>
<span><span class="co">#&gt; 18   GAPDH    ACTG1   CDC14A    IFI30</span></span>
<span><span class="co">#&gt; 19   ANXA1    H3F3B   PABPC1   FCER1G</span></span>
<span><span class="co">#&gt; 20   FXYD5     TPT1    FOXP1   TYROBP</span></span>
<span><span class="co">#&gt; 21   H3F3B HLA.DPB1     ACTB   FCGR3A</span></span>
<span><span class="co">#&gt; 22   EIF3E     GZMH    SORL1     TYMP</span></span>
<span><span class="co">#&gt; 23    BTG1    TRBC2   INPP4B    ACTG1</span></span>
<span><span class="co">#&gt; 24   TRBC2    KLRK1    SLFN5  S100A11</span></span>
<span><span class="co">#&gt; 25    PLP2     CD8A     KLF2 HLA.DPB1</span></span>
<span><span class="co">#&gt; 26  PABPC1  S100A10     KLF6      LYZ</span></span>
<span><span class="co">#&gt; 27  LGALS1 HLA.DPA1 HIST1H1E      VIM</span></span>
<span><span class="co">#&gt; 28   ITGB1    CRIP1  ANKRD11 HLA.DRB1</span></span>
<span><span class="co">#&gt; 29    JUNB    KLRD1   ATF7IP      CFD</span></span>
<span><span class="co">#&gt; 30    TSPO     GZMK  LRRFIP1    H3F3B</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="stacked_bar_plots">stacked_bar_plots<a class="anchor" aria-label="anchor" href="#stacked_bar_plots"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/stacked_bar_plots.html">stacked_bar_plots</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Primary"</span>,</span>
<span>  layers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seurat_clusters"</span>, <span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>  layer_orders <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Cohort <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HCV"</span>, <span class="st">"Healthy"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">colors</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"cNMF"</span>, show_counts <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-cnmf_files/figure-html/unnamed-chunk-14-1.png" class="r-plt" width="600"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS 26.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] yaml_2.3.11               sessioninfo_1.2.3        </span></span>
<span><span class="co">#&gt;  [3] matrixStats_1.5.0         rlang_1.1.6              </span></span>
<span><span class="co">#&gt;  [5] scCustomize_3.2.2         convert2anndata_0.2.0    </span></span>
<span><span class="co">#&gt;  [7] anndata_0.8.0             pheatmap_1.0.13          </span></span>
<span><span class="co">#&gt;  [9] ComplexHeatmap_2.25.2     harmony_1.2.4            </span></span>
<span><span class="co">#&gt; [11] Rcpp_1.1.0                Matrix_1.7-0             </span></span>
<span><span class="co">#&gt; [13] jsonlite_2.0.0            glue_1.8.0               </span></span>
<span><span class="co">#&gt; [15] stringr_1.6.0             treemap_2.4-4            </span></span>
<span><span class="co">#&gt; [17] RColorBrewer_1.1-3        eulerr_7.0.4             </span></span>
<span><span class="co">#&gt; [19] circlize_0.4.16           tibble_3.3.0             </span></span>
<span><span class="co">#&gt; [21] tidyr_1.3.1               dplyr_1.1.4              </span></span>
<span><span class="co">#&gt; [23] ggplot2_3.5.2             Seurat_5.1.0             </span></span>
<span><span class="co">#&gt; [25] SeuratObject_5.2.0        sp_2.2-0                 </span></span>
<span><span class="co">#&gt; [27] BioYourOwnBowl_0.0.0.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] ggtext_0.1.2                fs_1.6.6                   </span></span>
<span><span class="co">#&gt;   [3] spatstat.sparse_3.1-0       lubridate_1.9.4            </span></span>
<span><span class="co">#&gt;   [5] httr_1.4.7                  doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;   [7] tools_4.4.1                 sctransform_0.4.2          </span></span>
<span><span class="co">#&gt;   [9] backports_1.5.0             R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [11] lazyeval_0.2.2              uwot_0.2.4                 </span></span>
<span><span class="co">#&gt;  [13] GetoptLong_1.1.0            litedown_0.8               </span></span>
<span><span class="co">#&gt;  [15] withr_3.0.2                 gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [17] progressr_0.18.0            cli_3.6.5                  </span></span>
<span><span class="co">#&gt;  [19] Biobase_2.66.0              textshaping_1.0.4          </span></span>
<span><span class="co">#&gt;  [21] spatstat.explore_3.6-0      fastDummies_1.7.5          </span></span>
<span><span class="co">#&gt;  [23] labeling_0.4.3              sass_0.4.10                </span></span>
<span><span class="co">#&gt;  [25] spatstat.data_3.1-9         ggridges_0.5.7             </span></span>
<span><span class="co">#&gt;  [27] pbapply_1.7-4               pkgdown_2.2.0              </span></span>
<span><span class="co">#&gt;  [29] commonmark_2.0.0            systemfonts_1.3.1          </span></span>
<span><span class="co">#&gt;  [31] parallelly_1.45.1           rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;  [33] generics_0.1.4              shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;  [35] ica_1.0-3                   spatstat.random_3.4-3      </span></span>
<span><span class="co">#&gt;  [37] car_3.1-3                   ggbeeswarm_0.7.3           </span></span>
<span><span class="co">#&gt;  [39] S4Vectors_0.44.0            abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [41] lifecycle_1.0.4             snakecase_0.11.1           </span></span>
<span><span class="co">#&gt;  [43] carData_3.0-5               SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt;  [45] SparseArray_1.6.2           Rtsne_0.17                 </span></span>
<span><span class="co">#&gt;  [47] paletteer_1.6.0             promises_1.5.0             </span></span>
<span><span class="co">#&gt;  [49] crayon_1.5.3                miniUI_0.1.2               </span></span>
<span><span class="co">#&gt;  [51] lattice_0.22-6              cowplot_1.2.0              </span></span>
<span><span class="co">#&gt;  [53] pillar_1.11.1               knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [55] optparse_1.7.5              GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [57] rjson_0.2.23                future.apply_1.20.0        </span></span>
<span><span class="co">#&gt;  [59] codetools_0.2-20            leiden_0.4.3.1             </span></span>
<span><span class="co">#&gt;  [61] spatstat.univar_3.1-5       data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [63] vctrs_0.6.5                 png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [65] EnhancedVolcano_1.20.0      spam_2.11-1                </span></span>
<span><span class="co">#&gt;  [67] gtable_0.3.6                assertthat_0.2.1           </span></span>
<span><span class="co">#&gt;  [69] rematch2_2.1.2              cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [71] xfun_0.54                   S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt;  [73] mime_0.13                   survival_3.6-4             </span></span>
<span><span class="co">#&gt;  [75] SingleCellExperiment_1.28.1 iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [77] fitdistrplus_1.2-4          ROCR_1.0-11                </span></span>
<span><span class="co">#&gt;  [79] nlme_3.1-164                RcppAnnoy_0.0.22           </span></span>
<span><span class="co">#&gt;  [81] GenomeInfoDb_1.42.3         bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [83] irlba_2.3.5.1               vipor_0.4.7                </span></span>
<span><span class="co">#&gt;  [85] KernSmooth_2.23-24          otel_0.2.0                 </span></span>
<span><span class="co">#&gt;  [87] colorspace_2.1-2            BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt;  [89] ggrastr_1.0.2               tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [91] compiler_4.4.1              xml2_1.5.1                 </span></span>
<span><span class="co">#&gt;  [93] desc_1.4.3                  DelayedArray_0.32.0        </span></span>
<span><span class="co">#&gt;  [95] plotly_4.11.0               scales_1.4.0               </span></span>
<span><span class="co">#&gt;  [97] lmtest_0.9-40               digest_0.6.39              </span></span>
<span><span class="co">#&gt;  [99] goftest_1.2-3               spatstat.utils_3.2-0       </span></span>
<span><span class="co">#&gt; [101] rmarkdown_2.30              XVector_0.46.0             </span></span>
<span><span class="co">#&gt; [103] htmltools_0.5.9             pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [105] MatrixGenerics_1.18.1       fastmap_1.2.0              </span></span>
<span><span class="co">#&gt; [107] GlobalOptions_0.1.3         htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [109] UCSC.utils_1.2.0            shiny_1.12.0               </span></span>
<span><span class="co">#&gt; [111] farver_2.1.2                jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [113] zoo_1.8-14                  magrittr_2.0.4             </span></span>
<span><span class="co">#&gt; [115] Formula_1.2-5               GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">#&gt; [117] dotCall64_1.2               patchwork_1.3.2            </span></span>
<span><span class="co">#&gt; [119] reticulate_1.44.1           stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [121] zlibbioc_1.52.0             MASS_7.3-60.2              </span></span>
<span><span class="co">#&gt; [123] plyr_1.8.9                  parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [125] listenv_0.10.0              ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt; [127] forcats_1.0.1               deldir_2.0-4               </span></span>
<span><span class="co">#&gt; [129] splines_4.4.1               gridtext_0.1.5             </span></span>
<span><span class="co">#&gt; [131] tensor_1.5.1                igraph_2.2.1               </span></span>
<span><span class="co">#&gt; [133] ggpubr_0.6.2                spatstat.geom_3.6-1        </span></span>
<span><span class="co">#&gt; [135] ggsignif_0.6.4              markdown_2.0               </span></span>
<span><span class="co">#&gt; [137] RcppHNSW_0.6.0              reshape2_1.4.5             </span></span>
<span><span class="co">#&gt; [139] stats4_4.4.1                evaluate_1.0.5             </span></span>
<span><span class="co">#&gt; [141] renv_1.1.0                  ggprism_1.0.7              </span></span>
<span><span class="co">#&gt; [143] foreach_1.5.2               httpuv_1.6.16              </span></span>
<span><span class="co">#&gt; [145] RANN_2.6.2                  getopt_1.20.4              </span></span>
<span><span class="co">#&gt; [147] purrr_1.2.0                 polyclip_1.10-7            </span></span>
<span><span class="co">#&gt; [149] future_1.68.0               clue_0.3-66                </span></span>
<span><span class="co">#&gt; [151] scattermore_1.2             gridBase_0.4-7             </span></span>
<span><span class="co">#&gt; [153] janitor_2.2.1               broom_1.0.11               </span></span>
<span><span class="co">#&gt; [155] xtable_1.8-4                RSpectra_0.16-2            </span></span>
<span><span class="co">#&gt; [157] rstatix_0.7.3               later_1.4.4                </span></span>
<span><span class="co">#&gt; [159] viridisLite_0.4.2           ragg_1.5.0                 </span></span>
<span><span class="co">#&gt; [161] beeswarm_0.4.0              IRanges_2.40.1             </span></span>
<span><span class="co">#&gt; [163] cluster_2.1.6               timechange_0.3.0           </span></span>
<span><span class="co">#&gt; [165] globals_0.18.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Seurat: <a href="https://satijalab.org/seurat/" class="external-link uri">https://satijalab.org/seurat/</a>
</li>
<li>BioYourOwnBowl: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl" class="uri">https://github.com/june-zhang-bioinfo/BioYourOwnBowl</a>
</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by June Zhang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
