<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Customized Visualization • BioYourOwnBowl</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Customized Visualization">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioYourOwnBowl</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioYourOwnBowl-cnmf.html">Consensus Non-negative Matrix Factorization (cNMF) Results Exploration</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-main.html">Getting Started with BioYourOwnBowl</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-vis.html">Customized Visualization</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Customized Visualization</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/blob/HEAD/vignettes/BioYourOwnBowl-vis.Rmd"><code>vignettes/BioYourOwnBowl-vis.Rmd</code></a></small>
      <div class="hidden name"><code>BioYourOwnBowl-vis.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><strong>BioYourOwnBowl</strong> provides a comprehensive pipeline for
single-cell RNA-seq analysis using Seurat. This vignette demonstrates
the main workflow using example data from article: <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8376199/.The" class="external-link uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC8376199/.The</a>
seurat object is downsampled and has metadata included.</p>
<p>This vignette includes the advanced visualizations, e.g. : scaled
feature plots, violin plots (show bellies beyond zeros), heatmaps in 3
levels (cell-level, pseudobulk level and meta level), stacked bar plots,
splited dimplots, classic volcano plots, double volcano plots (show 2
volcanos at the same time), box plots with paired dots (connect paired
data point from one participant), venn plots showing DGE overlaps,
treemaps showing TCR clonotype distribution in different timepoint per
participant.</p>
<p>The plots will always be displayed in your RStudio. If a file name is
provided, it’ll be saved as a png file at the same time.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"june-zhang-bioinfo/BioYourOwnBowl"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<pre><code><span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; 'SeuratObject' was built under R 4.4.0 but the current version is</span></span>
<span><span class="co">#&gt; 4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI</span></span>
<span><span class="co">#&gt; for R may have changed</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'SeuratObject'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; circlize version 0.4.16</span></span>
<span><span class="co">#&gt; CRAN page: https://cran.r-project.org/package=circlize</span></span>
<span><span class="co">#&gt; Github page: https://github.com/jokergoo/circlize</span></span>
<span><span class="co">#&gt; Documentation: https://jokergoo.github.io/circlize_book/book/</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span><span class="co">#&gt; Gu, Z. circlize implements and enhances circular visualization</span></span>
<span><span class="co">#&gt;   in R. Bioinformatics 2014.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; This message can be suppressed by:</span></span>
<span><span class="co">#&gt;   suppressPackageStartupMessages(library(circlize))</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Matrix'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:tidyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand, pack, unpack</span></span>
<span><span class="co">#&gt; Loading required package: Rcpp</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; ComplexHeatmap version 2.25.2</span></span>
<span><span class="co">#&gt; Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span><span class="co">#&gt; Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span><span class="co">#&gt; Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite either one:</span></span>
<span><span class="co">#&gt; - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span><span class="co">#&gt; - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span><span class="co">#&gt;     genomic data. Bioinformatics 2016.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The new InteractiveComplexHeatmap package can directly export static </span></span>
<span><span class="co">#&gt; complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; This message can be suppressed by:</span></span>
<span><span class="co">#&gt;   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'pheatmap'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ComplexHeatmap':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     pheatmap</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'anndata'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:SeuratObject':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Layers</span></span>
<span><span class="co">#&gt; scCustomize v3.2.2</span></span>
<span><span class="co">#&gt; If you find the scCustomize useful please cite.</span></span>
<span><span class="co">#&gt; See 'samuel-marsh.github.io/scCustomize/articles/FAQ.html' for citation info.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'rlang'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:jsonlite':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     flatten, unbox</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'matrixStats'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     count</span></span>
<span><span class="co">#&gt; Warning: package 'yaml' was built under R version 4.4.3</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="prepare-example-data">Prepare Example Data<a class="anchor" aria-label="anchor" href="#prepare-example-data"></a>
</h2>
<p>Load the processed object from the main pipeline.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_dir_base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"BioYourOwnBowl"</span><span class="op">)</span></span>
<span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"GSE178756_processed.rds"</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">data_path</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 9453 features across 5893 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (9453 features, 2000 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#&gt;  2 dimensional reductions calculated: pca, umap</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-your-color-palette">Define Your Color Palette<a class="anchor" aria-label="anchor" href="#define-your-color-palette"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"#e6194B"</span>, </span>
<span>  <span class="st">"#86D3B2"</span>, </span>
<span>  <span class="st">"#5E8DBA"</span>, </span>
<span>  <span class="st">"#F7A131"</span>, </span>
<span>  <span class="st">"#911eb4"</span>, </span>
<span>  <span class="st">"#46f0f0"</span>, </span>
<span>  <span class="st">"#fabebe"</span>, </span>
<span>  <span class="st">"#bcf60c"</span>, </span>
<span>  <span class="st">"#B88C6D"</span>,</span>
<span>  <span class="st">"#4d88e8"</span>, </span>
<span>  <span class="st">"#FFFB47"</span>, </span>
<span>  <span class="st">"#A1A74A"</span>,</span>
<span>  <span class="st">"#7BB1A4"</span>, </span>
<span>  <span class="st">"#b15928"</span>,</span>
<span>  <span class="st">"#C493D4"</span>, </span>
<span>  <span class="st">"#f58231"</span>,</span>
<span>  <span class="st">"#0072B2"</span>,</span>
<span>  <span class="st">"#D55E00"</span>, </span>
<span>  <span class="st">"#009E73"</span>,</span>
<span>  <span class="st">"#CC79A7"</span>,</span>
<span>  <span class="st">"#F0E442"</span> </span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-selected-features">Define Selected Features<a class="anchor" aria-label="anchor" href="#define-selected-features"></a>
</h2>
<p>Provide specific genes of interest.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"CD8A"</span>, <span class="st">"CD8B"</span>,</span>
<span>  <span class="st">"CD3D"</span>, <span class="st">"CD3E"</span>, <span class="st">"CD3G"</span>,</span>
<span>  <span class="st">"GZMB"</span>, <span class="st">"GZMA"</span>, <span class="st">"GZMH"</span>,</span>
<span>  <span class="st">"PRF1"</span>, <span class="st">"NKG7"</span>, <span class="st">"GNLY"</span>,</span>
<span>  <span class="st">"IL2RA"</span>, <span class="st">"TNFRSF9"</span>, <span class="st">"CD69"</span>,</span>
<span>  <span class="st">"MKI67"</span>, <span class="st">"TOP2A"</span>,</span>
<span>  <span class="st">"PDCD1"</span>, <span class="st">"LAG3"</span>, <span class="st">"TIGIT"</span>,</span>
<span>  <span class="st">"CTLA4"</span>, <span class="st">"HAVCR2"</span>,</span>
<span>  <span class="st">"CCR7"</span>, <span class="st">"SELL"</span>, <span class="st">"TCF7"</span>,</span>
<span>  <span class="st">"LEF1"</span>, <span class="st">"IL7R"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="feature-plots">Feature plots<a class="anchor" aria-label="anchor" href="#feature-plots"></a>
</h3>
<p>This function generates 2 feature plot, one has original scaling (all
cells have color), one has customized scaling, controlled by argument
<code>percentile</code>. Only the cells having values beyond this
<code>percentile</code> will have color. The higher the percentile, the
more focused the feature plots.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/features_plots.html">features_plots</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD4"</span><span class="op">)</span>,</span>
<span>  n_col <span class="op">=</span> <span class="fl">3</span>, </span>
<span>  percentile <span class="op">=</span> <span class="fl">0.95</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `FetchData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; NOTE: <span style="color: #00BB00;">FeaturePlot_scCustom</span> uses a specified `na_cutoff` when plotting to</span></span>
<span><span class="co">#&gt; color cells with no expression as background color separate from color scale.</span></span>
<span><span class="co">#&gt; Please ensure `na_cutoff` value is appropriate for feature being plotted.</span></span>
<span><span class="co">#&gt; Default setting is appropriate for use when plotting from 'RNA' assay.</span></span>
<span><span class="co">#&gt; When `na_cutoff` not appropriate (e.g., module scores) set to NULL to</span></span>
<span><span class="co">#&gt; plot all cells in gradient color palette.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; -----This message will be shown once per session.-----</span></span>
<span><span class="co">#&gt; $original</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-7-1.png" class="r-plt" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $scaled</span></span></code></pre>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-7-2.png" class="r-plt" width="700"></p>
</div>
<div class="section level3">
<h3 id="violin-plots">Violin plots<a class="anchor" aria-label="anchor" href="#violin-plots"></a>
</h3>
<p>This function generates violin plots with box plot and dots
indicating the data distribution. The advantage is to see the bellies
beyond the zeros, which seurat built-in violin plot function fails to
do. It can also perform statistical tests and show the significance.
<code>add_stats = T</code> will trigger the tests on all binary
combinations, or you can indicates the comparison of interests.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/violin_plots.html">violin_plots</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD4"</span><span class="op">)</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  file_name <span class="op">=</span> <span class="st">"expression_violins.png"</span>, binwidth <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-8-1.png" class="r-plt" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># seurat violin</span></span>
<span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"seurat_clusters"</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD4"</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `PackageCheck()` was deprecated in SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `rlang::check_installed()` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-8-2.png" class="r-plt" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># statistical test for all comparisons</span></span>
<span><span class="fu"><a href="../reference/violin_plots.html">violin_plots</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD4"</span>, <span class="st">"TOX"</span>, <span class="st">"TCF7"</span><span class="op">)</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Time_point"</span>,</span>
<span>  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Normal"</span><span class="op">)</span>,</span>
<span>  add_stats <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"t.test"</span>,</span>
<span>  binwidth <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-8-3.png" class="r-plt" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># statistical test for specified comparisons</span></span>
<span><span class="va">my_comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"W4"</span>, <span class="st">"Normal"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/violin_plots.html">violin_plots</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD3D"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD4"</span>, <span class="st">"TOX"</span>, <span class="st">"TCF7"</span><span class="op">)</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Time_point"</span>,</span>
<span>  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Normal"</span><span class="op">)</span>,</span>
<span>  add_stats <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  comparisons <span class="op">=</span> <span class="va">my_comparisons</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  binwidth <span class="op">=</span> <span class="fl">0.05</span>, colors <span class="op">=</span> <span class="va">colors</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-8-4.png" class="r-plt" width="700"></p>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>There are 3 types of heatmaps having different granularity:
cell-level, pseudobulk-level and meta-level.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cell level</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/heatmap_cell_level.html">heatmap_cell_level</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="va">selected_features</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  colors <span class="op">=</span> <span class="va">colors</span>,</span>
<span>  fontsize <span class="op">=</span> <span class="fl">6</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Centering and scaling data matrix</span></span>
<span><span class="co">#&gt; Warning: Different features in new layer data than already exists for</span></span>
<span><span class="co">#&gt; scale.data</span></span>
<span><span class="co">#&gt; Automatically calculated height: 6 inches for 23 genes</span></span>
<span><span class="co">#&gt; `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">#&gt; 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">#&gt; setting TRUE/FALSE to it.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="co">#&gt; 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Set `ht_opt$message = FALSE` to turn off this message.</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">hm</span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-9-1.png" class="r-plt" width="500"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># pseudobulk level</span></span>
<span><span class="fu"><a href="../reference/heatmap_pseudobulk.html">heatmap_pseudobulk</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">seurat_obj</span>, </span>
<span>                   features <span class="op">=</span> <span class="va">selected_features</span>, </span>
<span>                   groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Post_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span>,</span>
<span>                   idents <span class="op">=</span> <span class="st">"Time_point"</span>, </span>
<span>                   pseudobulk_column <span class="op">=</span> <span class="st">"orig.ident"</span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-9-2.png" class="r-plt" width="500"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># meta level</span></span>
<span><span class="fu"><a href="../reference/heatmap_meta.html">heatmap_meta</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features <span class="op">=</span> <span class="va">selected_features</span>, </span>
<span>             groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Post_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span>, </span>
<span>             idents <span class="op">=</span> <span class="st">"Time_point"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Normalizing layer: counts</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-9-3.png" class="r-plt" width="500"></p>
</div>
<div class="section level3">
<h3 id="stacked-bar-plots">Stacked bar plots<a class="anchor" aria-label="anchor" href="#stacked-bar-plots"></a>
</h3>
<p>This function is versatile. You can use it to visualize the cluster
distribution, or simply to check meta data. It can take up to 3 layers:
the 1st layer represents the individual bar (the lowest level), the 2nd
and 3rd are used to group the bars. You can specify the order of each
layers (e.g. chronologically).</p>
<p>Use <code>show_counts</code> = TRUE to see the cell counts per
bar.</p>
<p>For the best layout, when you input 3 layers, combine_stacked_bars()
is required to combine them.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># one layer</span></span>
<span><span class="fu"><a href="../reference/stacked_bar_plots.html">stacked_bar_plots</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  layers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>  colors <span class="op">=</span> <span class="va">colors</span>, </span>
<span>  show_counts <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-10-1.png" class="r-plt" width="500"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># two layers</span></span>
<span><span class="fu"><a href="../reference/stacked_bar_plots.html">stacked_bar_plots</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Time_point"</span>,</span>
<span>  layers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"orig.ident"</span>, <span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>  colors <span class="op">=</span> <span class="va">colors</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-10-2.png" class="r-plt" width="500"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># three layers</span></span>
<span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stacked_bar_plots.html">stacked_bar_plots</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  layers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"orig.ident"</span>,<span class="st">"Time_point"</span>, <span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>  layer_orders <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    orig.ident <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_order.html" class="external-link">str_sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    Time_point <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span>, <span class="st">"Post_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span>,</span>
<span>    Cohort <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HCV"</span>, <span class="st">"Healthy"</span><span class="op">)</span></span>
<span>    <span class="op">)</span>,</span>
<span>  colors <span class="op">=</span> <span class="va">colors</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Input 3 layers. Use combine_stacked_bars() to combine them.</span></span>
<span></span>
<span><span class="fu"><a href="../reference/combine_stacked_bars.html">combine_stacked_bars</a></span><span class="op">(</span>plot_list <span class="op">=</span> <span class="va">plot_list</span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-10-3.png" class="r-plt" width="500"></p>
</div>
<div class="section level3">
<h3 id="dimplot-to-check-batch-effect-or-status-movement">Dimplot to check batch effect or status movement<a class="anchor" aria-label="anchor" href="#dimplot-to-check-batch-effect-or-status-movement"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/dimplots_split.html">dimplots_split</a></span><span class="op">(</span><span class="va">seurat_obj</span>, </span>
<span>               split_var <span class="op">=</span> <span class="st">"orig.ident"</span>, </span>
<span>               group_var <span class="op">=</span> <span class="st">"seurat_clusters"</span>, </span>
<span>               colors <span class="op">=</span> <span class="va">colors</span><span class="op">)</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-11-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level3">
<h3 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/volcano_plots.html">volcano_plots</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Time_point"</span>,</span>
<span>  ident1 <span class="op">=</span> <span class="st">"Pre_Tx"</span>,</span>
<span>  ident2 <span class="op">=</span> <span class="st">"W4"</span>,</span>
<span>  show_labels <span class="op">=</span> <span class="va">selected_features</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"CD8 T cells"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Performing differential expression: Pre_Tx vs. W4</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `GetAssayData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">Seurat</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/satijalab/seurat/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; For a (much!) faster implementation of the Wilcoxon Rank Sum Test,</span></span>
<span><span class="co">#&gt; (default method for FindMarkers) please install the presto package</span></span>
<span><span class="co">#&gt; --------------------------------------------</span></span>
<span><span class="co">#&gt; install.packages('devtools')</span></span>
<span><span class="co">#&gt; devtools::install_github('immunogenomics/presto')</span></span>
<span><span class="co">#&gt; --------------------------------------------</span></span>
<span><span class="co">#&gt; After installation of presto, Seurat will automatically use the more </span></span>
<span><span class="co">#&gt; efficient implementation (no further action necessary).</span></span>
<span><span class="co">#&gt; This message will be shown once per session</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-12-1.png"><!-- --></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/volcano_plots.html">volcano_plots</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Time_point"</span>,</span>
<span>  ident1 <span class="op">=</span> <span class="st">"W4"</span>,</span>
<span>  ident2 <span class="op">=</span> <span class="st">"Post_Tx"</span>,</span>
<span>  show_labels <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"CD8 T cells"</span>, log2fc_cutoff <span class="op">=</span> <span class="fl">0.7</span>, label_size <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Performing differential expression: W4 vs. Post_Tx</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-12-2.png"><!-- --></p>
</div>
<div class="section level3">
<h3 id="double-volcano">Double volcano<a class="anchor" aria-label="anchor" href="#double-volcano"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create your comparison table</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ident1 <span class="op">=</span> <span class="st">"W4"</span>, ident2 <span class="op">=</span> <span class="st">"Pre_Tx"</span>, ident3 <span class="op">=</span> <span class="st">"Pre_Tx"</span>, ident4 <span class="op">=</span> <span class="st">"Post_Tx"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Time_point"</span></span>
<span><span class="fu"><a href="../reference/double_volcano.html">double_volcano</a></span><span class="op">(</span><span class="va">seurat_obj</span>, </span>
<span>               <span class="va">df</span>, </span>
<span>               output_dir <span class="op">=</span> <span class="st">"/Users/zz005/Documents/github/R/2dim_log2fc"</span>,</span>
<span>               lfc_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span>, </span>
<span>               pval_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>               selected_genes <span class="op">=</span> <span class="va">selected_features</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text()`).</span></span>
<span><span class="co">#&gt; Processed row 1: W4 vs. Pre_Tx vs. Pre_Tx vs. Post_Tx</span></span>
<span><span class="co">#&gt; Warning: Removed 3 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text()`).</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-13-1.png" class="r-plt" width="600"></p>
</div>
<div class="section level3">
<h3 id="boxplot-with-paired-dots">Boxplot with paired dots<a class="anchor" aria-label="anchor" href="#boxplot-with-paired-dots"></a>
</h3>
<p>Since there is no paired timepoints in any of the participant in this
object, no dots are connected.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/boxplot_paired_points.html">boxplot_paired_points</a></span><span class="op">(</span><span class="va">seurat_obj</span>, gene <span class="op">=</span> <span class="st">"TOX"</span>, </span>
<span>                      timepoint_col <span class="op">=</span> <span class="st">"Time_point"</span>, </span>
<span>                      participant_col <span class="op">=</span> <span class="st">"orig.ident"</span>, </span>
<span>                      pre_label <span class="op">=</span> <span class="st">"Pre_Tx"</span>, </span>
<span>                      post_label <span class="op">=</span> <span class="st">"Post_Tx"</span><span class="op">)</span></span>
<span><span class="co">#&gt; $plot</span></span></code></pre></div>
<p><img src="BioYourOwnBowl-vis_files/figure-html/unnamed-chunk-14-1.png" class="r-plt" width="400"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $data</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3 variables: Patient &lt;chr&gt;, Timepoint &lt;fct&gt;, Value &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="venn-plots">Venn plots<a class="anchor" aria-label="anchor" href="#venn-plots"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Specify the comparisons of interest</span></span>
<span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span><span class="st">"Pre_Tx vs. W4"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"W4"</span><span class="op">)</span>,</span>
<span><span class="st">"Pre_Tx vs. Normal"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pre_Tx"</span>, <span class="st">"Normal"</span><span class="op">)</span>,</span>
<span><span class="st">"W4 vs. Normal"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"W4"</span>, <span class="st">"Normal"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make venn plot</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/venn_plots.html">venn_plots</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat_obj</span>,</span>
<span>  comparisons <span class="op">=</span> <span class="va">comparisons</span>,</span>
<span>  idents <span class="op">=</span> <span class="st">"Time_point"</span>,</span>
<span>  log2fc_threshold <span class="op">=</span> <span class="fl">0.58</span>,</span>
<span>  min_pct <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  main <span class="op">=</span> <span class="st">"Upregulated Genes"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Performing differential expression comparisons...</span></span>
<span><span class="co">#&gt;   Pre_Tx vs. W4: Pre_Tx vs W4</span></span>
<span><span class="co">#&gt;     Found 5 genes</span></span>
<span><span class="co">#&gt;   Pre_Tx vs. Normal: Pre_Tx vs Normal</span></span>
<span><span class="co">#&gt;     Found 44 genes</span></span>
<span><span class="co">#&gt;   W4 vs. Normal: W4 vs Normal</span></span>
<span><span class="co">#&gt;     Found 54 genes</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="treemaps">Treemaps<a class="anchor" aria-label="anchor" href="#treemaps"></a>
</h3>
<p>This dataset doesn’t have TCR information, so we use simulated data
for illustration purpose.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_cells</span> <span class="op">&lt;-</span> <span class="fl">1500</span></span>
<span></span>
<span><span class="va">participants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"P"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">epitopes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Epitope1"</span>, <span class="st">"Epitope2"</span>, <span class="st">"Epitope3"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  participant <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">participants</span>, <span class="va">n_cells</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  Specificity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">epitopes</span>, <span class="va">n_cells</span>, replace <span class="op">=</span> <span class="cn">TRUE</span>, prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.3</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  raw_clonotype_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"clono"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">80</span><span class="op">)</span>, <span class="va">n_cells</span>, replace <span class="op">=</span> <span class="cn">TRUE</span>, prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Exponential.html" class="external-link">rexp</a></span><span class="op">(</span><span class="fl">80</span>, rate <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cell"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_cells</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add some NA clonotypes for realism</span></span>
<span><span class="va">meta</span><span class="op">$</span><span class="va">raw_clonotype_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">n_cells</span>, <span class="fl">40</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span></span>
<span><span class="co"># ---------------------------</span></span>
<span><span class="co"># Create minimal Seurat object</span></span>
<span><span class="co"># ---------------------------</span></span>
<span><span class="co"># Simulate counts matrix</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="va">n_cells</span> <span class="op">*</span> <span class="fl">300</span>, lambda <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>  nrow <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  ncol <span class="op">=</span> <span class="va">n_cells</span>,</span>
<span>  dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">300</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span>, meta.data <span class="op">=</span> <span class="va">meta</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Data is of class matrix. Coercing to dgCMatrix.</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="../reference/tcr_treemaps.html">tcr_treemaps</a></span><span class="op">(</span><span class="va">object</span>, </span>
<span>             top_epitope <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Epitope2"</span>, <span class="st">"Epitope3"</span><span class="op">)</span>, </span>
<span>             min_cells <span class="op">=</span> <span class="fl">20</span>, </span>
<span>             output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>             color_palette <span class="op">=</span> <span class="va">colors</span><span class="op">)</span></span>
<span><span class="co">#&gt; Created 6 treemap plots in /Users/zz005/Documents/github/R/BioYourOwnBowl/vignettes</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS 26.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] yaml_2.3.11               sessioninfo_1.2.3        </span></span>
<span><span class="co">#&gt;  [3] matrixStats_1.5.0         rlang_1.1.6              </span></span>
<span><span class="co">#&gt;  [5] scCustomize_3.2.2         convert2anndata_0.2.0    </span></span>
<span><span class="co">#&gt;  [7] anndata_0.8.0             pheatmap_1.0.13          </span></span>
<span><span class="co">#&gt;  [9] ComplexHeatmap_2.25.2     harmony_1.2.4            </span></span>
<span><span class="co">#&gt; [11] Rcpp_1.1.0                Matrix_1.7-0             </span></span>
<span><span class="co">#&gt; [13] jsonlite_2.0.0            glue_1.8.0               </span></span>
<span><span class="co">#&gt; [15] stringr_1.6.0             treemap_2.4-4            </span></span>
<span><span class="co">#&gt; [17] RColorBrewer_1.1-3        eulerr_7.0.4             </span></span>
<span><span class="co">#&gt; [19] circlize_0.4.16           tibble_3.3.0             </span></span>
<span><span class="co">#&gt; [21] tidyr_1.3.1               dplyr_1.1.4              </span></span>
<span><span class="co">#&gt; [23] ggplot2_3.5.2             Seurat_5.1.0             </span></span>
<span><span class="co">#&gt; [25] SeuratObject_5.2.0        sp_2.2-0                 </span></span>
<span><span class="co">#&gt; [27] BioYourOwnBowl_0.0.0.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RcppAnnoy_0.0.22            splines_4.4.1              </span></span>
<span><span class="co">#&gt;   [3] later_1.4.4                 polyclip_1.10-7            </span></span>
<span><span class="co">#&gt;   [5] janitor_2.2.1               fastDummies_1.7.5          </span></span>
<span><span class="co">#&gt;   [7] lifecycle_1.0.4             rstatix_0.7.3              </span></span>
<span><span class="co">#&gt;   [9] doParallel_1.0.17           globals_0.18.0             </span></span>
<span><span class="co">#&gt;  [11] lattice_0.22-6              MASS_7.3-60.2              </span></span>
<span><span class="co">#&gt;  [13] backports_1.5.0             magrittr_2.0.4             </span></span>
<span><span class="co">#&gt;  [15] plotly_4.11.0               sass_0.4.10                </span></span>
<span><span class="co">#&gt;  [17] rmarkdown_2.30              jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [19] httpuv_1.6.16               otel_0.2.0                 </span></span>
<span><span class="co">#&gt;  [21] sctransform_0.4.2           spam_2.11-1                </span></span>
<span><span class="co">#&gt;  [23] spatstat.sparse_3.1-0       EnhancedVolcano_1.20.0     </span></span>
<span><span class="co">#&gt;  [25] reticulate_1.44.1           cowplot_1.2.0              </span></span>
<span><span class="co">#&gt;  [27] pbapply_1.7-4               lubridate_1.9.4            </span></span>
<span><span class="co">#&gt;  [29] abind_1.4-8                 zlibbioc_1.52.0            </span></span>
<span><span class="co">#&gt;  [31] Rtsne_0.17                  GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [33] purrr_1.2.0                 BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt;  [35] GenomeInfoDbData_1.2.13     IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [37] S4Vectors_0.44.0            ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt;  [39] irlba_2.3.5.1               listenv_0.10.0             </span></span>
<span><span class="co">#&gt;  [41] spatstat.utils_3.2-0        goftest_1.2-3              </span></span>
<span><span class="co">#&gt;  [43] RSpectra_0.16-2             spatstat.random_3.4-3      </span></span>
<span><span class="co">#&gt;  [45] fitdistrplus_1.2-4          parallelly_1.45.1          </span></span>
<span><span class="co">#&gt;  [47] pkgdown_2.2.0               leiden_0.4.3.1             </span></span>
<span><span class="co">#&gt;  [49] codetools_0.2-20            getopt_1.20.4              </span></span>
<span><span class="co">#&gt;  [51] DelayedArray_0.32.0         tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [53] shape_1.4.6.1               UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt;  [55] farver_2.1.2                stats4_4.4.1               </span></span>
<span><span class="co">#&gt;  [57] spatstat.explore_3.6-0      GetoptLong_1.1.0           </span></span>
<span><span class="co">#&gt;  [59] Formula_1.2-5               progressr_0.18.0           </span></span>
<span><span class="co">#&gt;  [61] ggridges_0.5.7              survival_3.6-4             </span></span>
<span><span class="co">#&gt;  [63] iterators_1.0.14            systemfonts_1.3.1          </span></span>
<span><span class="co">#&gt;  [65] polylabelr_0.3.0            foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [67] tools_4.4.1                 ragg_1.5.0                 </span></span>
<span><span class="co">#&gt;  [69] ica_1.0-3                   gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [71] SparseArray_1.6.2           xfun_0.54                  </span></span>
<span><span class="co">#&gt;  [73] MatrixGenerics_1.18.1       GenomeInfoDb_1.42.3        </span></span>
<span><span class="co">#&gt;  [75] withr_3.0.2                 fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [77] digest_0.6.39               timechange_0.3.0           </span></span>
<span><span class="co">#&gt;  [79] R6_2.6.1                    mime_0.13                  </span></span>
<span><span class="co">#&gt;  [81] ggprism_1.0.7               textshaping_1.0.4          </span></span>
<span><span class="co">#&gt;  [83] colorspace_2.1-2            Cairo_1.7-0                </span></span>
<span><span class="co">#&gt;  [85] scattermore_1.2             tensor_1.5.1               </span></span>
<span><span class="co">#&gt;  [87] spatstat.data_3.1-9         generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [89] renv_1.1.0                  data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [91] httr_1.4.7                  htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [93] S4Arrays_1.6.0              uwot_0.2.4                 </span></span>
<span><span class="co">#&gt;  [95] pkgconfig_2.0.3             gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [97] lmtest_0.9-40               SingleCellExperiment_1.28.1</span></span>
<span><span class="co">#&gt;  [99] XVector_0.46.0              htmltools_0.5.9            </span></span>
<span><span class="co">#&gt; [101] carData_3.0-5               dotCall64_1.2              </span></span>
<span><span class="co">#&gt; [103] clue_0.3-66                 scales_1.4.0               </span></span>
<span><span class="co">#&gt; [105] Biobase_2.66.0              png_0.1-8                  </span></span>
<span><span class="co">#&gt; [107] optparse_1.7.5              snakecase_0.11.1           </span></span>
<span><span class="co">#&gt; [109] spatstat.univar_3.1-5       knitr_1.50                 </span></span>
<span><span class="co">#&gt; [111] rstudioapi_0.17.1           reshape2_1.4.5             </span></span>
<span><span class="co">#&gt; [113] rjson_0.2.23                nlme_3.1-164               </span></span>
<span><span class="co">#&gt; [115] cachem_1.1.0                zoo_1.8-14                 </span></span>
<span><span class="co">#&gt; [117] GlobalOptions_0.1.3         KernSmooth_2.23-24         </span></span>
<span><span class="co">#&gt; [119] vipor_0.4.7                 parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [121] miniUI_0.1.2                ggrastr_1.0.2              </span></span>
<span><span class="co">#&gt; [123] desc_1.4.3                  pillar_1.11.1              </span></span>
<span><span class="co">#&gt; [125] vctrs_0.6.5                 RANN_2.6.2                 </span></span>
<span><span class="co">#&gt; [127] ggpubr_0.6.2                promises_1.5.0             </span></span>
<span><span class="co">#&gt; [129] car_3.1-3                   xtable_1.8-4               </span></span>
<span><span class="co">#&gt; [131] cluster_2.1.6               paletteer_1.6.0            </span></span>
<span><span class="co">#&gt; [133] beeswarm_0.4.0              evaluate_1.0.5             </span></span>
<span><span class="co">#&gt; [135] cli_3.6.5                   compiler_4.4.1             </span></span>
<span><span class="co">#&gt; [137] crayon_1.5.3                ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [139] future.apply_1.20.0         labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [141] rematch2_2.1.2              forcats_1.0.1              </span></span>
<span><span class="co">#&gt; [143] ggbeeswarm_0.7.3            plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [145] fs_1.6.6                    stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [147] gridBase_0.4-7              viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [149] deldir_2.0-4                assertthat_0.2.1           </span></span>
<span><span class="co">#&gt; [151] lazyeval_0.2.2              spatstat.geom_3.6-1        </span></span>
<span><span class="co">#&gt; [153] RcppHNSW_0.6.0              patchwork_1.3.2            </span></span>
<span><span class="co">#&gt; [155] future_1.68.0               shiny_1.12.0               </span></span>
<span><span class="co">#&gt; [157] SummarizedExperiment_1.36.0 ROCR_1.0-11                </span></span>
<span><span class="co">#&gt; [159] broom_1.0.11                igraph_2.2.1               </span></span>
<span><span class="co">#&gt; [161] bslib_0.9.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Seurat: <a href="https://satijalab.org/seurat/" class="external-link uri">https://satijalab.org/seurat/</a>
</li>
<li>BioYourOwnBowl: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl" class="uri">https://github.com/june-zhang-bioinfo/BioYourOwnBowl</a>
</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by June Zhang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>




  inst/pkgdown/images.css

  </body>
</html>
