<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index â€¢ BioYourOwnBowl</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/journal/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioYourOwnBowl</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/BioYourOwnBowl-cnmf.html">Consensus Non-negative Matrix Factorization (cNMF) Results Exploration</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-main.html">Getting Started with BioYourOwnBowl</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-vis.html">Customized Visualization</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="anchor_features.html">anchor_features()</a></code> </p>
        </td>
        <td><p>Find integration anchors and set variable features</p></td>
      </tr><tr><td>
          <p><code><a href="boxplot_paired_points.html">boxplot_paired_points()</a></code> </p>
        </td>
        <td><p>Plot Paired Gene Expression with Links and Faceting</p></td>
      </tr><tr><td>
          <p><code><a href="clustering.html">clustering()</a></code> </p>
        </td>
        <td><p>Automated parameter sweep for Seurat clustering</p></td>
      </tr><tr><td>
          <p><code><a href="cnmf_bar_plots.html">cnmf_bar_plots()</a></code> </p>
        </td>
        <td><p>Plot Top N Usage Columns From cNMF Gene Spectra</p></td>
      </tr><tr><td>
          <p><code><a href="cnmf_gene_ranks.html">cnmf_gene_ranks()</a></code> </p>
        </td>
        <td><p>Search Gene Ranks Across cNMF Programs</p></td>
      </tr><tr><td>
          <p><code><a href="cnmf_top_programs.html">cnmf_top_programs()</a></code> </p>
        </td>
        <td><p>Add top cNMF programs to Seurat metadata with usage cutoff</p></td>
      </tr><tr><td>
          <p><code><a href="cnmf_umaps.html">cnmf_umaps()</a></code> </p>
        </td>
        <td><p>Plot UMAP Usage Maps From cNMF Usage Matrix</p></td>
      </tr><tr><td>
          <p><code><a href="combine_stacked_bars.html">combine_stacked_bars()</a></code> </p>
        </td>
        <td><p>Combine a List of Cluster Distribution Plots</p></td>
      </tr><tr><td>
          <p><code><a href="compare_cnmf_programs.html">compare_cnmf_programs()</a></code> </p>
        </td>
        <td><p>Compare Overlap Between Two cNMF Program Sets</p></td>
      </tr><tr><td>
          <p><code><a href="density_plot.html">density_plot()</a></code> </p>
        </td>
        <td><p>Generate Python Code for Scanpy UMAP Density Plots</p></td>
      </tr><tr><td>
          <p><code><a href="dimplots_split.html">dimplots_split()</a></code> </p>
        </td>
        <td><p>Save Split DimPlots from a Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="dotplots_pdf.html">dotplots_pdf()</a></code> </p>
        </td>
        <td><p>Generate DotPlot PDFs for a Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="dotplots_png.html">dotplots_png()</a></code> </p>
        </td>
        <td><p>Generate and save a Seurat DotPlot dynamically</p></td>
      </tr><tr><td>
          <p><code><a href="double_volcano.html">double_volcano()</a></code> </p>
        </td>
        <td><p>Generate 2D Log2 Fold-Change Comparison Plots Between Identity Pairs</p></td>
      </tr><tr><td>
          <p><code><a href="export_top_pc_genes.html">export_top_pc_genes()</a></code> </p>
        </td>
        <td><p>Export top genes per principal component</p></td>
      </tr><tr><td>
          <p><code><a href="features_plots.html">features_plots()</a></code> </p>
        </td>
        <td><p>Plot Features with Dynamic Layout and Cutoffs</p></td>
      </tr><tr><td>
          <p><code><a href="filter_genes_by_expression.html">filter_genes_by_expression()</a></code> </p>
        </td>
        <td><p>Filter Genes by Minimum Cell Expression</p></td>
      </tr><tr><td>
          <p><code><a href="heatmap_cell_level.html">heatmap_cell_level()</a></code> </p>
        </td>
        <td><p>Generate Clustered Heatmap for Seurat Object</p></td>
      </tr><tr><td>
          <p><code><a href="heatmap_meta.html">heatmap_meta()</a></code> </p>
        </td>
        <td><p>Generate Cell-Type-Level Expression Heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="heatmap_pseudobulk.html">heatmap_pseudobulk()</a></code> </p>
        </td>
        <td><p>Generate Patient-Level Expression Heatmap with Z-scores</p></td>
      </tr><tr><td>
          <p><code><a href="optimize_single_cell.html">optimize_single_cell()</a></code> </p>
        </td>
        <td><p>Optimize Single Cell RNA-seq Analysis</p></td>
      </tr><tr><td>
          <p><code><a href="plot_mean_variance.html">plot_mean_variance()</a></code> </p>
        </td>
        <td><p>Plot mean-variance relationship with feature highlighting</p></td>
      </tr><tr><td>
          <p><code><a href="prepare_h5ad.html">prepare_h5ad()</a></code> </p>
        </td>
        <td><p>Prepare a Seurat Object for Export to AnnData (.h5ad)</p></td>
      </tr><tr><td>
          <p><code><a href="preprocess_obj.html">preprocess_obj()</a></code> </p>
        </td>
        <td><p>Preprocess Seurat Object.</p></td>
      </tr><tr><td>
          <p><code><a href="remove_low_quality_clusters.html">remove_low_quality_clusters()</a></code> </p>
        </td>
        <td><p>Remove low-quality clusters based on feature count thresholds</p></td>
      </tr><tr><td>
          <p><code><a href="select_PCs.html">select_PCs()</a></code> </p>
        </td>
        <td><p>Select principal components based on variance explained</p></td>
      </tr><tr><td>
          <p><code><a href="select_marker_genes_score.html">select_marker_genes_score()</a></code> </p>
        </td>
        <td><p>Select marker genes based on product score</p></td>
      </tr><tr><td>
          <p><code><a href="setup_python.html">setup_python()</a></code> </p>
        </td>
        <td><p>Set up Python environment for BioYourOwnBowl</p></td>
      </tr><tr><td>
          <p><code><a href="stacked_bar_plots.html">stacked_bar_plots()</a></code> </p>
        </td>
        <td><p>Plot Cluster Distribution Across Metadata Layers</p></td>
      </tr><tr><td>
          <p><code><a href="tcr_treemaps.html">tcr_treemaps()</a></code> </p>
        </td>
        <td><p>Generate Treemaps of TCR Clonotypes for Selected Epitopes</p></td>
      </tr><tr><td>
          <p><code><a href="venn_plots.html">venn_plots()</a></code> </p>
        </td>
        <td><p>Create Venn Diagram from Differential Expression Comparisons</p></td>
      </tr><tr><td>
          <p><code><a href="violin_plots.html">violin_plots()</a></code> </p>
        </td>
        <td><p>Create Violin Plot with Boxplot and Dotplot Overlay</p></td>
      </tr><tr><td>
          <p><code><a href="volcano_plots.html">volcano_plots()</a></code> </p>
        </td>
        <td><p>Create Volcano Plot from Differential Expression Analysis</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by June Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

