<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Optimize Single Cell RNA-seq Analysis — optimize_single_cell • BioYourOwnBowl</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/journal/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Optimize Single Cell RNA-seq Analysis — optimize_single_cell"><meta property="og:description" content="This function performs a comprehensive single-cell RNA-seq analysis pipeline including
quality control, normalization, dimensionality reduction, clustering optimization, and
visualization. It automatically saves parameters, session info, and generates multiple
QC and analysis plots."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioYourOwnBowl</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/BioYourOwnBowl-cnmf.html">Consensus Non-negative Matrix Factorization (cNMF) Results Exploration</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-main.html">Getting Started with BioYourOwnBowl</a>
    </li>
    <li>
      <a href="../articles/BioYourOwnBowl-vis.html">Customized Visualization</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Optimize Single Cell RNA-seq Analysis</h1>
    <small class="dont-index">Source: <a href="https://github.com/june-zhang-bioinfo/BioYourOwnBowl/blob/HEAD/R/seurat_utils.R"><code>R/seurat_utils.R</code></a></small>
    <div class="hidden name"><code>optimize_single_cell.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function performs a comprehensive single-cell RNA-seq analysis pipeline including
quality control, normalization, dimensionality reduction, clustering optimization, and
visualization. It automatically saves parameters, session info, and generates multiple
QC and analysis plots.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">optimize_single_cell</span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>  vf.method <span class="op">=</span> <span class="st">"vst"</span>,</span>
<span>  model.use <span class="op">=</span> <span class="st">"linear"</span>,</span>
<span>  pseudobulk_var <span class="op">=</span> <span class="st">"patient_tp_mutation"</span>,</span>
<span>  pseudo_cutoff <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mt.cutoff <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  harmony <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  harmony_vars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>  improved_diff_quantile <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  k_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span><span class="op">)</span>,</span>
<span>  resolution_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.3</span>, <span class="fl">0.1</span><span class="op">)</span>,</span>
<span>  min.dist <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">17</span>,</span>
<span>  gene_filter <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  regress.out <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  clusters_min <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  clusters_max <span class="op">=</span> <span class="fl">13</span>,</span>
<span>  min.pct <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  lower_bound_param <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  higher_bound_param <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  cluster_distribution_layers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cluster_distribution_layer_orders <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  features <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-object">object<a class="anchor" aria-label="anchor" href="#arg-object"></a></dt>
<dd><p>A Seurat object.</p></dd>


<dt id="arg-normalization-method">normalization.method<a class="anchor" aria-label="anchor" href="#arg-normalization-method"></a></dt>
<dd><p>Method for normalization. Default is "LogNormalize". Please refer to Seurat::NormalizeData() for more options.</p></dd>


<dt id="arg-vf-method">vf.method<a class="anchor" aria-label="anchor" href="#arg-vf-method"></a></dt>
<dd><p>Method for variable feature selection. Options include "vst", "dispersion",
"pseudobulk", "sct", or "anchor". Default is "vst".</p></dd>


<dt id="arg-model-use">model.use<a class="anchor" aria-label="anchor" href="#arg-model-use"></a></dt>
<dd><p>Model to use for scaling. Default is "linear". Please refer to Seurat::ScaleData() for more options.</p></dd>


<dt id="arg-pseudobulk-var">pseudobulk_var<a class="anchor" aria-label="anchor" href="#arg-pseudobulk-var"></a></dt>
<dd><p>Variable name for pseudobulk aggregation. Default is "patient_tp_mutation".</p></dd>


<dt id="arg-pseudo-cutoff">pseudo_cutoff<a class="anchor" aria-label="anchor" href="#arg-pseudo-cutoff"></a></dt>
<dd><p>Minimum number of cells required for pseudobulk calculation. Default is 30.</p></dd>


<dt id="arg-colors">colors<a class="anchor" aria-label="anchor" href="#arg-colors"></a></dt>
<dd><p>Named vector of colors for plotting. Default is NULL.</p></dd>


<dt id="arg-mt-cutoff">mt.cutoff<a class="anchor" aria-label="anchor" href="#arg-mt-cutoff"></a></dt>
<dd><p>Maximum percent mitochondrial content threshold for cell filtering. Default is 25.</p></dd>


<dt id="arg-nfeatures">nfeatures<a class="anchor" aria-label="anchor" href="#arg-nfeatures"></a></dt>
<dd><p>Number of variable features to select. Default is 2000.</p></dd>


<dt id="arg-harmony">harmony<a class="anchor" aria-label="anchor" href="#arg-harmony"></a></dt>
<dd><p>Logical indicating whether to run Harmony batch correction. Default is FALSE.</p></dd>


<dt id="arg-harmony-vars">harmony_vars<a class="anchor" aria-label="anchor" href="#arg-harmony-vars"></a></dt>
<dd><p>Character vector of variable names to use for Harmony correction. e.g. run, experiment, patient. Default is NULL.</p></dd>


<dt id="arg-reduction">reduction<a class="anchor" aria-label="anchor" href="#arg-reduction"></a></dt>
<dd><p>Dimensionality reduction method to use ("pca" or "harmony"). Default is "pca".</p></dd>


<dt id="arg-improved-diff-quantile">improved_diff_quantile<a class="anchor" aria-label="anchor" href="#arg-improved-diff-quantile"></a></dt>
<dd><p>Quantile threshold for PC selection. Lower values select more PCs. Default is 0.6.</p></dd>


<dt id="arg-k-range">k_range<a class="anchor" aria-label="anchor" href="#arg-k-range"></a></dt>
<dd><p>Vector of k values for nearest neighbor calculations. Default is c(10, 20, 30).</p></dd>


<dt id="arg-resolution-range">resolution_range<a class="anchor" aria-label="anchor" href="#arg-resolution-range"></a></dt>
<dd><p>Vector of resolution values for clustering. Default is seq(0.1, 0.3, 0.1).</p></dd>


<dt id="arg-min-dist">min.dist<a class="anchor" aria-label="anchor" href="#arg-min-dist"></a></dt>
<dd><p>Minimum distance parameter for UMAP. Default is 0.4.</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>Random seed for reproducibility. Default is 17.</p></dd>


<dt id="arg-gene-filter">gene_filter<a class="anchor" aria-label="anchor" href="#arg-gene-filter"></a></dt>
<dd><p>Minimum number of cells a gene must be expressed in to be retained. Default is 50.</p></dd>


<dt id="arg-regress-out">regress.out<a class="anchor" aria-label="anchor" href="#arg-regress-out"></a></dt>
<dd><p>Variables to regress out during scaling, e.g. proliterating genes. Default is NULL.</p></dd>


<dt id="arg-clusters-min">clusters_min<a class="anchor" aria-label="anchor" href="#arg-clusters-min"></a></dt>
<dd><p>Minimum number of clusters expected. Default is 4.</p></dd>


<dt id="arg-clusters-max">clusters_max<a class="anchor" aria-label="anchor" href="#arg-clusters-max"></a></dt>
<dd><p>Maximum number of clusters expected. Default is 13.</p></dd>


<dt id="arg-min-pct">min.pct<a class="anchor" aria-label="anchor" href="#arg-min-pct"></a></dt>
<dd><p>Minimum percentage expressed required for a gene to become DEG. Default is 0.3.</p></dd>


<dt id="arg-lower-bound-param">lower_bound_param<a class="anchor" aria-label="anchor" href="#arg-lower-bound-param"></a></dt>
<dd><p>Lower bound parameter for low-quality cluster removal. To be more stringent, decrease this parameter. Default is 1.</p></dd>


<dt id="arg-higher-bound-param">higher_bound_param<a class="anchor" aria-label="anchor" href="#arg-higher-bound-param"></a></dt>
<dd><p>Upper bound parameter for low-quality cluster removal. To be more stringent, increase this parameter. Default is 2.</p></dd>


<dt id="arg-cluster-distribution-layers">cluster_distribution_layers<a class="anchor" aria-label="anchor" href="#arg-cluster-distribution-layers"></a></dt>
<dd><p>Character vector of metadata columns for cluster distribution plots. Default is NULL.</p></dd>


<dt id="arg-cluster-distribution-layer-orders">cluster_distribution_layer_orders<a class="anchor" aria-label="anchor" href="#arg-cluster-distribution-layer-orders"></a></dt>
<dd><p>List of factor level orders for distribution layers. Default is NULL.</p></dd>


<dt id="arg-out-dir">out_dir<a class="anchor" aria-label="anchor" href="#arg-out-dir"></a></dt>
<dd><p>Output directory for saving results. Default is current working directory.</p></dd>


<dt id="arg-features">features<a class="anchor" aria-label="anchor" href="#arg-features"></a></dt>
<dd><p>Data frame or list of selected features for visualization. Default is NULL.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A processed Seurat object with clustering and dimensionality reduction completed.
The object is also saved as "object.rds" in the output directory.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The function performs the following steps:</p><ol><li><p>Creates output directory and logs parameters and session info</p></li>
<li><p>Filters genes by expression threshold</p></li>
<li><p>Processes variable features using specified method</p></li>
<li><p>Runs PCA and optionally Harmony batch correction</p></li>
<li><p>Performs initial clustering at high resolution for QC</p></li>
<li><p>Removes low-quality clusters and high mitochondrial content cells</p></li>
<li><p>Re-processes variable features after QC</p></li>
<li><p>Selects optimal number of PCs</p></li>
<li><p>Runs clustering across multiple k and resolution parameters</p></li>
<li><p>Generates comprehensive visualizations including cluster distributions, dotplots, and heatmaps</p></li>
</ol></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Basic usage</span></span></span>
<span class="r-in"><span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu">optimize_single_cell</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">my_seurat_object</span>,</span></span>
<span class="r-in"><span>  out_dir <span class="op">=</span> <span class="st">"results/analysis_v1"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># With Harmony batch correction</span></span></span>
<span class="r-in"><span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu">optimize_single_cell</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">my_seurat_object</span>,</span></span>
<span class="r-in"><span>  harmony <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  harmony_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"batch"</span>, <span class="st">"donor"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  out_dir <span class="op">=</span> <span class="st">"results/harmony_analysis"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Custom clustering parameters</span></span></span>
<span class="r-in"><span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu">optimize_single_cell</span><span class="op">(</span></span></span>
<span class="r-in"><span>  object <span class="op">=</span> <span class="va">my_seurat_object</span>,</span></span>
<span class="r-in"><span>  k_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">25</span>, <span class="fl">35</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  resolution_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span>, <span class="fl">0.2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  clusters_min <span class="op">=</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>  clusters_max <span class="op">=</span> <span class="fl">15</span>,</span></span>
<span class="r-in"><span>  out_dir <span class="op">=</span> <span class="st">"results/custom_clustering"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by June Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

